From 4c04b59bc372f030ed274e75203425da10270869 Mon Sep 17 00:00:00 2001
From: Anupam Ghosh <anupam.ghosh@siemens.com>
Date: Thu, 16 Jan 2020 17:15:22 +0530
Subject: [PATCH] fix(nomos) : segfault for large offset value

Signed-off-by: Anupam Ghosh <anupam.ghosh@siemens.com>

Upstream-Status: Backport [https://github.com/fossology/fossology/commit/4c04b59bc372f030ed274e75203425da10270869]
---
 src/nomos/agent/parse.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/nomos/agent/parse.c b/src/nomos/agent/parse.c
index f50ecc4c6..b79387626 100755
--- a/src/nomos/agent/parse.c
+++ b/src/nomos/agent/parse.c
@@ -10237,8 +10237,10 @@ int match3(int base, char *buf, int score, int save, int isML, int isPS)
 void saveLicenseParagraph(char *mtext, int isML, int isPS, int entireBuf)
 {
   char *cp;
-  char *start = mtext;
+  char *start;
+  int index=0;
   int len;
+  start =  copyString(mtext, MTAG_TEXTPARA);
   if(!start)
   {
     LOG_FATAL("called saveLicenseParagraph without text")
@@ -10258,13 +10260,15 @@ void saveLicenseParagraph(char *mtext, int isML, int isPS, int entireBuf)
       len = cur.regm.rm_eo + 80;
     } else {
       len = cur.regm.rm_eo + 130 - cur.regm.rm_so;
-      start += cur.regm.rm_so - 50;
+      index = cur.regm.rm_so - 50;
     }
     cur.licPara = memAlloc(len + 9, MTAG_TEXTPARA);
     (void) strcpy(cur.licPara, "... ");
-    (void) strncpy(cur.licPara + 4, start, len);
+    (void) strncpy(cur.licPara + 4, start + index, len);
     (void) strcpy(cur.licPara + len + 4, " ...");
+    memFree(start, MTAG_TEXTPARA);
   }
+
   /*
    * Convert double-line-feed chars ("\r" and "\n" combos) to a single "\n"
    */
-- 
2.20.1

