From a1b287e06209ad3f3559e789f744130144f610fb Mon Sep 17 00:00:00 2001
From: Anupam Ghosh <anupam.ghosh@siemens.com>
Date: Thu, 16 Jan 2020 17:37:49 +0530
Subject: [PATCH] fix(nomos) : CC-BY-SA identification

Signed-off-by: Anupam Ghosh <anupam.ghosh@siemens.com>

Upstream-Status: Backport [https://github.com/fossology/fossology/commit/a1b287e06209ad3f3559e789f744130144f610fb]
---
 src/nomos/agent/STRINGS.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nomos/agent/STRINGS.in b/src/nomos/agent/STRINGS.in
index c193b72f1..705c47c94 100755
--- a/src/nomos/agent/STRINGS.in
+++ b/src/nomos/agent/STRINGS.in
@@ -7833,7 +7833,7 @@ k
 #
 %ENTRY% _TITLE_CC_BY_SA
 %KEY% "a(gree|ttribut|vailabl|uthor)"
-%STR% "creative commons attribution[ -]share[ -]alike"
+%STR% "creative commons attribution[ -]share.?alike"
 #
 %ENTRY% _TITLE_CC_BY_SA_10
 %KEY% "a(gree|ttribut|vailabl|uthor)"
-- 
2.20.1

