From 9b327e80b33ac5f6282064a98c58df8b61ef3d7b Mon Sep 17 00:00:00 2001
From: rlintu <raino.lintulampi@bittium.com>
Date: Fri, 23 Oct 2020 15:04:42 +0300
Subject: [PATCH] Nomos: New licenses from SPDX 3.10 added. Lots of other
 corrections.

Upstream-Status: Backport [https://github.com/fossology/fossology/commit/9b327e80b33ac5f6282064a98c58df8b61ef3d7b]
---

diff --git a/src/nomos/agent/STRINGS.in b/src/nomos/agent/STRINGS.in
index bd16db232..b2fb01c24 100755
--- a/src/nomos/agent/STRINGS.in
+++ b/src/nomos/agent/STRINGS.in
@@ -788,6 +788,18 @@
 %KEY% "permi[st]"
 %STR% "permission to use copy modify and distribute this software for any purpose is hereby granted without fee provided that the above copyright notice and this permission notice appear in all copies"
 #
+%ENTRY% _LT_blessing
+%KEY% "legal"
+%STR% "in place of a legal notice here is a blessing"
+#
+%ENTRY% _LT_BlueOak_100_Purpose
+%KEY% "software"
+%STR% "gives everyone as much permission to work with this software as possible"
+#
+%ENTRY% _LT_BlueOak_100_Acceptance
+%KEY% "licen[cs]"
+%STR% "rules of this license are both obligations under"
+#
 %ENTRY% _LT_BOGUSTMPL
 %KEY% "licen[cs]"
 %STR% "This file is r?e?-?distributed under the same licen[cs]e as the package PACKAGE"
@@ -891,6 +903,10 @@
 %KEY% "licen[cs]"
 %STR% "subject to the terms and conditions of the bsd license"
 #
+%ENTRY% _LT_BSDref15
+%KEY% "licen[cs]"
+%STR% "is bsd.licensed"
+#
 %ENTRY% _LT_BSDSTYLEref1
 %KEY% "\<under\>"
 %STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|code) =SOME= under =SOME= (BSD[ -]style|BSD[ -]type|BSD-?ish|BSD[ -]like) licen[cs]e"
@@ -1020,7 +1036,7 @@
 #
 %ENTRY% _LT_BSD_CLAUSE_1
 %KEY% "copyright"
-%STR% "redistributions? of source code must retain the (above )?copyright notice (unmodified )?this list of conditions and the following disclaimer"
+%STR% "redistributions? of source code must retain the (above |original author.s )?copyright notice (unmodified )?this list of conditions and the following disclaimer"
 #
 %ENTRY% _LT_BSD_CLAUSE_1_DISCLAIMER
 %KEY% "copyright"
@@ -1028,7 +1044,7 @@
 #
 %ENTRY% _LT_BSD_CLAUSE_2
 %KEY% "copyright"
-%STR% "redistributions? in binary form must reproduc[et] the (above )?copyright notice this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution"
+%STR% "redistributions? in binary form must reproduc[et] the (above |original author.s )?copyright notice this list of conditions and the following disclaimer (listed in this license )?in the documentation and ?/or other materials provided with the distribution"
 #
 %ENTRY% _LT_BSD_CLAUSE_3
 %KEY% "software"
@@ -1050,6 +1066,10 @@
 %KEY% "licen[cs]"
 %STR% "no express or implied licenses to any =FEW= patent rights are granted by this license"
 #
+%ENTRY% _LT_BSD_CLAUSE_OPEN_MPI
+%KEY% "so(ftware|urce)"
+%STR% "no reassurances that the source code provided does not infringe any patent"
+#
 %ENTRY% _LT_BSD_CLAUSE_PATENT
 %KEY% "licen[cs]"
 %STR% "patent license to make have made use offer to sell sell import and otherwise transfer this software"
@@ -1108,7 +1128,7 @@
 #
 %ENTRY% _LT_CC_ref
 %KEY% =NULL=
-%STR% "(distribu|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used?) =SOME= \<creative ?commons\>"
+%STR% "(distribu|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used?|see) =SOME= \<creative ?commons\>"
 #
 # _LT_CC_BY_SA is for v 2.5
 %ENTRY% _LT_CC_BY_SA_25
@@ -1128,11 +1148,7 @@
 %ENTRY% _LT_CC_BY_ND_30
 %KEY% "derogatory"
 %STR% "You must not distort, mutilate, modify or take other derogatory"
-#####
-# DELETED: ... you a world-?wide royalty-free non-?exclusive perpetual
-# for the duration of the applicable copyright licen[cs]e to exercise
-# the rights in the work as stated below
-#####
+#
 %ENTRY% _LT_CCPL
 %KEY% "(condit|vers)ion"
 %STR% "subject to the terms and conditions of this (public )?licen[cs]e =FEW= licen[cs]or hereby grants"
@@ -1384,9 +1400,13 @@
 %KEY% "so(ftware|urce)"
 %STR% "free software you (can|may) r?e?-?distribute (it a|a)nd/or modify it under the terms of =SOME= as published by DARPA"
 #
-%ENTRY% _LT_DARPA_COUGAAR
+%ENTRY% _LT_DARPA_COUGAAR_1
 %KEY% "so(ftware|urce)"
 %STR% "free software you (can|may) r?e?-?distribute (it a|a)nd/or modify it under the terms of the cougaar open source licen[cs]e as published by DARPA"
+#
+%ENTRY% _LT_DARPA_COUGAAR_2
+%KEY% "so(ftware|urce)"
+%STR% "this product includes software developed in part by support from the defense advanced research project agency"
 #####
 # DELETED: ... any party from selling or giving away the software
 # as a component of an aggregate software distribution containing
@@ -1501,7 +1521,7 @@
 #
 %ENTRY% _LT_DUAL_LICENSE_9
 %KEY% "licen[cs]"
-%STR% "(is|was|are|were) (\<dual\>|multiple)[ -]licensed"
+%STR% "(is|was|are|were) (\<dual\>|multiple)[ -]license"
 #
 %ENTRY% _LT_DUAL_LICENSE_10
 %KEY% "licen[cs]"
@@ -1620,7 +1640,7 @@
 #
 %ENTRY% _LT_DUAL_LICENSE_38
 %KEY% "licen[cs]"
-%STR% "you (may|can) choose between two licenses"
+%STR% "you (may|can) choose (between two|either) licenses?"
 #
 %ENTRY% _LT_DUAL_LICENSE_39
 %KEY% "licen[cs]"
@@ -1665,7 +1685,7 @@
 #
 %ENTRY% _LT_DUAL_LICENSE_49
 %KEY% "licen[cs]"
-%STR% "spdx-license-identifier.{1,100}[[:space:]]OR[[:space:]]"
+%STR% "spdx-license-identifier[-_:\+\.\(\)[:alnum:][:blank:]]+[[:blank:]]OR[[:blank:]]"
 #
 %ENTRY% _LT_SUBSEQUENT_VERSION
 %KEY% "licen[cs]"
@@ -2088,7 +2108,7 @@
 #
 %ENTRY% _LT_FREE_51
 %KEY% "distribut"
-%STR% "copying and distribution of this file with or without modification are permitted (in any medium without royalty )?provided"
+%STR% "copying and distribution of this file with or without modification are permitted provided"
 #
 %ENTRY% _LT_FREE_52
 %KEY% "distribut"
@@ -2330,6 +2350,10 @@
 %KEY% "distribut"
 %STR% "you may freely use modify and/or r?e?distribute this file"
 #
+%ENTRY% _LT_FREE_111
+%KEY% "\<free\>"
+%STR% "free for all purposes without any express guarantee"
+#
 %ALIAS% _FREE_last_other
 %ALIAS% _FREECLAIM_last
 #
@@ -2415,15 +2439,19 @@
 %ENTRY% _LT_FSF_5
 %KEY% "distribut"
 %STR% "and you are welcome to r?e?-?distribut(e|e copies of) it under certain conditions"
-#####
-# DELETED: free =SOME= ... to copy distribute and modify it
-#####
+#
 %ENTRY% _LT_FSFref1
 %KEY% "permi[st]"
 %STR% "the free software foundation gives unlimited permission"
-#####
-# DELETED: ... you preserve this copyright notice and permission notice
-#####
+#
+%ENTRY% _LT_FSFAP
+%KEY% "distribut"
+%STR% "copying and distribution of this file with or without modification are permitted in any medium without royalty provided"
+#
+%ENTRY% _LT_FSFUL
+%KEY% "distribut"
+%STR% "gives unlimited permission to copy distribute and modify it"
+#
 %ENTRY% _LT_LG
 %KEY% "licen[cs]"
 %STR% "License agreement"
@@ -2685,7 +2713,7 @@
 #
 %ENTRY% _LT_GPL_NAMED2
 %KEY% "gpl"
-%STR% "(under|terms|releas) ==SOME== [^al]gpl"
+%STR% "(under|terms|releas) =SOME= [^al]gpl"
 #
 # To catch things like "this is GPL".
 %ENTRY% _LT_GPL_NAMED3
@@ -3112,6 +3140,14 @@
 %ENTRY% _LT_HP_PROPRIETARY_3
 %KEY% "so(ftware|urce)"
 %STR% "hp grants you a licen[cs]e to use one copy of the software use means storing loading installing executing or displaying the software you may not modify the software or disable any licen[cs]ing"
+#
+%ENTRY% _LT_HPND_1
+%KEY% "distribut"
+%STR% "not be used in advertising or publicity pertaining to distribution"
+#
+%ENTRY% _LT_HPND_2
+%KEY% "so(ftware|urce)"
+%STR% "no representations about the suitability of this software"
 #####
 # DELETED: ... =ANY= is hereby granted provided that the above
 # copyright notices? =SOME= appear in all copies
@@ -3442,6 +3478,14 @@
 %KEY% "licen[cs]"
 %STR% "subject to =FEW= licensing requirements contained herein you may =FEW= use the specifications and materials =FEW= IPTC"
 #
+%ENTRY% _LT_ISSL_1
+%KEY% "distribut"
+%STR% "you may use and redistribute =SOME= without modification"
+#
+%ENTRY% _LT_ISSL_2
+%KEY% "licen[cs]"
+%STR% "Intel grants you a world-wide royalty-free non-exclusive license under patents"
+#
 %ENTRY% _LT_ITU_T_1
 %KEY% "licen[cs]"
 %STR% "source code and all of its derivations is subject to the itu-t general public license"
@@ -3662,7 +3706,7 @@
 #####
 %ENTRY% _LT_LGPLref2
 %KEY% "\<under\>"
-%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|\<use\>|code) =SOME= under =SOME= (\<lgpl|lesser GNU (general p|p)ublic licen[cs]e|GNU (lesser|library) (gpl|(general p|p)ublic licen[cs]e))"
+%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|\<use\>|code|published) =SOME= under =SOME= (\<lgpl|lesser GNU (general p|p)ublic licen[cs]e|GNU (lesser|library) (gpl|(general p|p)ublic licen[cs]e))"
 #
 %ENTRY% _LT_LGPLref3
 %KEY% "(condit|vers)ion"
@@ -3721,7 +3765,7 @@
 #
 %ENTRY% _LT_LGPL3ref
 %KEY% "\<under\>"
-%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|code) =SOME= under =SOME= (lgpl|lesser GNU general public licen[cs]e|GNU (lesser|library) (gpl|(general p|p)ublic licen[cs]e))(([v-]?| ?version) ?)3"
+%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|code|published) =SOME= under =SOME= (lgpl|lesser GNU general public licen[cs]e|GNU (lesser|library) (gpl|(general p|p)ublic licen[cs]e))(([v-]| ?version) ?)3"
 #
 %ENTRY% _LT_LGPL3_ref_later
 %KEY% "licen[cs]"
@@ -3737,7 +3781,7 @@
 # To catch things like "this is LGPL".
 %ENTRY% _LT_LGPL_NAMED2
 %KEY% "lgpl"
-%STR% "[^-_]\<lgpl\>"
+%STR% "[^-_]\<lgpl(v(2|3))?\>"
 #
 %ENTRY% _LT_LGPL_OR
 %NOCHECK%
@@ -3752,10 +3796,18 @@
 %KEY% "rar" 
 %STR% "re-create the rar compression algorithm"
 #
+%ENTRY% _LT_libmng_2007_1
+%KEY% =NULL=
+%STR% "result from the use of the MNG Library"
+#
 %ENTRY% _LT_libmng_2007_Clause_1
 %KEY% "software"
 %STR% "you must not claim that you wrote the original software"
 #
+%ENTRY% _LT_libpng_20_Clause_1
+%KEY% "software"
+%STR% "if you use this software =FEW= acknowledgment =FEW= would be appreciated"
+#
 %ENTRY% _LT_linking_exception_1
 %KEY% "permi[st]"
 %STR% "in addition to the permissions in =FEW= general public license =FEW= gives you unlimited permission to link the compiled version of this file (into combinations )?with other programs"
@@ -4008,7 +4060,7 @@
 #
 %ENTRY% _LT_MITref7
 %KEY% "licen[cs]"
-%STR% "licen[cs]ed?:? (type )?\<mit\>"
+%STR% "licen[cs]ed?:? (type |is )?\<mit\>"
 #
 %ENTRY% _LT_MITref8
 %KEY% "licen[cs]"
@@ -4199,7 +4251,7 @@
 #
 %ENTRY% _LT_MSCORP_PLref3
 %KEY% "licen[cs]"
-%STR% "subject to (the )?microsoft (permissive|public) licen[cs]e"
+%STR% "subject to (the |terms and conditions of the )?microsoft (permissive|public) licen[cs]e"
 #####
 # DELETED: ... your products maintaining your products or enhancing the
 # interoperability of your products with the software and specifically
@@ -4232,6 +4284,10 @@
 %KEY% "software"
 %STR% "the name MTL shall not be used to endorse or promote products derived from this software"
 #
+%ENTRY% _LT_MulanPSL_10
+%KEY% "software"
+%STR% "distribution of the software shall be subject to Mulan PSL (version |v)1"
+#
 %ENTRY% _LT_MULTICORP_1
 %KEY% "licen[cs]"
 %STR% "each agree to gran(t|t (to y|y)ou) a licen[cs]e under royalty-free and otherwise reasonable non-discriminatory terms and conditions to their respective essential patent claims"
@@ -4303,13 +4359,18 @@
 # and distribute the original software or portions thereof with or
 # without modifications and/or as part of a larger work
 #####
+%ENTRY% _LT_NIST_1
+%KEY% "distribut"
+%STR% "you may use copy and distribute copies of the software =SOME= keep intact this entire notice"
+#
+%ENTRY% _LT_NIST_2
+%KEY% "distribut"
+%STR% "you may improve modify and create derivative works of the software =SOME= distribute such modifications or works"
+#
 %ENTRY% _LT_NOKIA
 %KEY% "licen[cs]"
 %STR% "under copyrights licen[cs]able by Nokia"
-#####
-# DELETED: permission is granted to copy and create derivative works for ...
-# is preserved in all copies of source code or included in human readable form
-#####
+#
 %ENTRY% _LT_NONCOMMERCIAL_1
 %KEY% "commerc"
 %STR% "any non-?commercial purpose provided th(is|ese) copyright notice"
@@ -4370,6 +4431,10 @@
 %KEY% "permi[st]"
 %STR% "permission to use copy modify and distribute this software and its documentation for any purpose (with or|and) without fee is hereby granted provided that the above copyright notice appears in all copies and that both the copyright notice and this permission notice appear in supporting documentation"
 #
+%ENTRY% _LT_NTP_0
+%KEY% "so(ftware|urce)"
+%STR% "advertising or =FEW= distribution of the software without =FEW= permission"
+#
 %ENTRY% _LT_NVIDIA
 %KEY% "(©|\(c\)|copyright|\<c\>[^+:]|&copy)"
 %STR% "hereby granted a non-?exclusive,? royalty-free copyright (and design patent)? licen[cs]e"
@@ -4412,6 +4477,10 @@
 %ENTRY% _LT_ODL
 %KEY% "licen[cs]"
 %STR% "subject to all of the terms and conditions of this Open Directory licen[cs]e"
+#
+%ENTRY% _LT_OGC
+%KEY% "licen[cs]"
+%STR% "OGC work =SOME= is being provided by the copyright holders under the following license"
 #####
 # DELETED: ... on with or for operating systems which are not open
 # source programs
@@ -4515,7 +4584,7 @@
 #
 %ENTRY% _LT_OpenSSL_exception_2
 %KEY% "openssl"
-%STR% "as (an additional|a special) (exemption|exception) =SOME= permission to link =FEW= with the (OpenSSL project.s )?OpenSSL librar(y|ies)"
+%STR% "as (an additional|a(nother)? special) (exemption|exception) =SOME= permission to link =SOME= with (any version of )?the (OpenSSL project.s )?OpenSSL librar(y|ies)"
 #
 %ENTRY% _LT_openvpn_openssl_exception
 %KEY% "openssl"
@@ -4796,7 +4865,6 @@
 #####
 %ENTRY% _LT_PUBDOM_CC
 %KEY% "p(atent|roduc|rofit|roject|ublic)"
-#%STR% "copyright-only dedication =SOME= or public[ -]domain cert"
 %STR% "to the best of his knowledge the work of authorship identified is in the \<public\>[ -]domain of the country from which the work is published"
 #
 %ENTRY% _LT_PURDUE
@@ -4942,7 +5010,7 @@
 #
 %ENTRY% _LT_PYTHONref3
 %KEY% "licen[cs]"
-%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|code) =SOME= python =FEW= licen[cs]e"
+%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|code) =SOME= \<python =FEW= licen[cs]e"
 #
 %ENTRY% _LT_PYTHONSTYLEref
 %KEY% "licen[cs]"
@@ -5346,6 +5414,18 @@
 %ENTRY% _LT_SNIAref
 %KEY% "licen[cs]"
 %STR% "(distribu|develop|offere?d?|released?|licen[cs]ed?|available|protected|provided|subject|comes) =SOME= (snia|storage network industry) =FEW= public licen[cs]e"
+#
+%ENTRY% _LT_Spencer_86_94_CLAUSE_1
+%KEY% "software"
+%STR% "author is not responsible for the consequences of use of this software"
+#
+%ENTRY% _LT_Spencer_94_CLAUSE_2
+%KEY% "source"
+%STR% "since few users ever read (the )?sources credits must appear in the documentation"
+#
+%ENTRY% _LT_Spencer_99
+%KEY% "software"
+%STR% "given credit for this package in the documentation of software"
 #####
 # DELETED: permission is granted to make and distribute verbatim copies
 # of this =SOME= ...
@@ -5368,6 +5448,14 @@
 %KEY% "distribut"
 %STR% "use and r?e?-?distribution for any purpose are granted as long as all copyright notices are retained r?e?-?distribution with modification is allowed"
 #
+%ENTRY% _LT_SSH
+%KEY% =NULL=
+%STR% "it must be called by a name other than \"?ssh\"? or \"?secure shell\"?"
+#
+%ENTRY% _LT_SSH_OpenSSH
+%KEY% "permi[st]"
+%STR% "legal status of this program is some combination of all these permissions and restrictions"
+#
 %ENTRY% _LT_SSLEAY
 %KEY% "(©|\(c\)|copyright|\<c\>[^+:]|&copy)"
 %STR% "copyright remains Eric Young =FEW= as such any copyright notices in the code are not to be removed"
@@ -5546,7 +5634,7 @@
 #
 %ENTRY% _LT_TCL
 %KEY% "software"
-%STR% "this software is copyrighted by the Regents of the University of California Sun Microsystem =FEW= Scriptics Corporation =SOME= and other parties =SOME= apply to all files associated with the software unless explicitly disclaimed in individual files"
+%STR% "this software is copyrighted by the Regents of the University of California Sun Microsystem =SOME= and other parties =SOME= apply to all files associated with the software unless explicitly disclaimed in individual files"
 #####
 # DELETED: ... distribute this software and its documentation for
 # any purpose and without fee provided that
@@ -5738,11 +5826,11 @@
 %KEY% "licen[cs]"
 %STR% "provided under specific written licen[cs]e from the University of Washington and may only"
 #
-%ENTRY% _LT_VIM_1
+%ENTRY% _LT_Vim_1
 %KEY% "distribut"
 %STR% "if you make changes to Vim yourself you must clearly describe in the distribution how to contact you"
 #
-%ENTRY% _LT_VIM_2
+%ENTRY% _LT_Vim_2
 %KEY% "restrict"
 %STR% "no restrictions on distributing unmodified copies of Vim except that they must include this licen[cs]e text"
 #
@@ -5778,43 +5866,23 @@
 %ENTRY% _LT_W3C_1
 %KEY% "permi[st]"
 %STR% "permission to use copy and/?o?r? distribute the contents"
-#####
-# DELETED: ... or royalty is hereby granted provided that you include
-# the following on all copies of the software and documentation or
-# portions thereof including modifications that you make
-#####
+#
 %ENTRY% _LT_W3C_2
 %KEY% "permi[st]"
 %STR% "permission to use copy modify and/?o?r? distribute this software and its documentatio(n|n with or without modification) for any purpose (and w|with or w|w)ithout fee"
-#####
-# VERY similar to W3C_2 above, except 's/use copy/copy/'
-#####
-# DELETED: ... and its documentation with or without modification for
-# any purpose and without fee or royalty is hereby granted provided
-# that you include the following on all copies
-#####
+#
 %ENTRY% _LT_W3C_3
 %KEY% "permi[st]"
-%STR% "permission to copy modify and/?o?r? distribute this software"
-#####
-# DELETED: ... code =ANY= for any purpose without fee subject to the
-# following
-#####
+%STR% "permission to copy modify and/?o?r? distribute this (software|work)"
+#
 %ENTRY% _LT_W3C_4
 %KEY% "permi[st]"
 %STR% "permission is hereby granted to use copy modify and/?o?r? distribute this source"
-#####
-# DELETED: this document or the W3C document from which this statement
-# is linked
-#####
+#
 %ENTRY% _LT_W3C_5
 %KEY% "permi[st]"
 %STR% "permission to copy and/?o?r? distribute the contents of"
-#####
-# DELETED: permission to use copy modify and distribute the XHTML Basic
-# DTD and its accompanying documentation for any purpose and without fee
-# is hereby ...
-#####
+#
 %ENTRY% _LT_W3C_6
 %KEY% "(©|\(c\)|copyright|\<c\>[^+:]|&copy)"
 %STR% "granted in perpetuity provided that the above copyright notice and this paragraph appear in all copies"
@@ -5825,6 +5893,22 @@
 %KEY% "permi[st]"
 %STR% "permission to use copy and/?o?r? modify this software"
 #
+%ENTRY% _LT_W3C_8
+%KEY% "notice"
+%STR% "W3C software (and document )?short notice should be included"
+#
+%ENTRY% _LT_W3C_19980720
+%KEY% "notice"
+%STR% "notice of any changes or modifications to the W3C files"
+#
+%ENTRY% _LT_W3C_20021231
+%KEY% "notice"
+%STR% "notice of any changes or modifications to the files including the date"
+#
+%ENTRY% _LT_W3C_20150513
+%KEY% "notice"
+%STR% "notice of any changes or modifications through a copyright statement"
+#
 %ENTRY% _LT_W3Cref1
 %KEY% "licen[cs]"
 %STR% "based =SOME= w3c =SOME= software notice and licen[cs]e"
@@ -6278,7 +6362,7 @@ k
 #
 %ENTRY% _LT_SEE_COPYING_FINAL2
 %KEY% "\<(see|sell|sale|sold|charge)\>"
-%STR% "see the (file =FEW= \<copying\>|\<copying\> file) =SOME= (distribut|dir|copy|us(e|age)|mod|free soft)"
+%STR% "see the (file =FEW= \<copying\>|\<copying\> file) =SOME= (distribut|dir|copy|us(e|age)|mod|free soft|details)"
 #
 %ENTRY% _LT_SEE_COPYING_FINAL3
 %NOCHECK%
@@ -6444,7 +6528,7 @@ k
 #
 %ENTRY% _LT_SEE_LICENSE_18
 %KEY% "licen[cs]"
-%STR% "see (the )?license file =FEW= for licensing terms"
+%STR% "see (the )?license( file|\.?txt) =FEW= for licens(ing terms|e information)"
 #
 %ENTRY% _LT_SEE_LICENSE_19
 %KEY% "licen[cs]"
@@ -7297,6 +7381,10 @@ k
 %KEY% "so(ftware|urce)"
 %STR% "(©|\(c\)|copyright|\<c\>[^+:]|&copy) =YEAR= =SOME= sleepycat software"
 #
+%ENTRY% _CR_Spencer
+%KEY% "so(ftware|urce)"
+%STR% "(©|\(c\)|copyright|\<c\>[^+:]|&copy) =YEAR= =SOME= Henry Spencer"
+#
 %ENTRY% _CR_SPI
 %KEY% "so(ftware|urce)"
 %STR% "(©|\(c\)|copyright|\<c\>[^+:]|&copy) =YEAR= =SOME= (software in the public interest inc|\<spi\>)"
@@ -7623,19 +7711,27 @@ k
 %KEY% "\<ap(ache|ple)\>"
 %STR% "apache =FEW= licen[cs]e"
 #
-%ENTRY% _TITLE_COPYLEFT_NEXT_030
+%ENTRY% _TITLE_CERN_OHL_11
+%KEY% "licen[cs]"
+%STR% "CERN Open Hardware Licence (version |v)1\.?1"
+#
+%ENTRY% _TITLE_CERN_OHL_12
+%KEY% "licen[cs]"
+%STR% "CERN Open Hardware Licence (version |v)1\.?2"
+#
+%ENTRY% _TITLE_copyleft_next_030
 %KEY% "licen[cs]"
 %STR% "copyleft-next 0\.?3\.?0 =FEW= license release date 2013-05-16"
 #
-%ENTRY% _TITLE_COPYLEFT_NEXT_031
+%ENTRY% _TITLE_copyleft_next_031
 %KEY% "licen[cs]"
 %STR% "copyleft-next 0\.?3\.?1 =FEW= license release date 2016-04-29"
 #
-%ENTRY% _PHR_COPYLEFT_NEXT_PARA1
+%ENTRY% _PHR_copyleft_next_PARA1
 %KEY% "licen[cs]"
 %STR% "license grants no trademark license"
 #
-%ENTRY% _PHR_COPYLEFT_NEXT_PARA3
+%ENTRY% _PHR_copyleft_next_PARA3
 %KEY% "distribut"
 %STR% "conditions for distributing derived works outbound gpl compatibility"
 #
@@ -7687,6 +7783,10 @@ k
 %KEY% "licen[cs]"
 %STR% "bittorrent open source licen[cs]e (v|version )1\.?0"
 #
+%ENTRY% _TITLE_BlueOak_100
+%KEY% "licen[cs]"
+%STR% "Blue Oak Model License (v|version )1\.?0\.?0"
+#
 %ENTRY% _TITLE_BOOST
 %KEY% "licen[cs]"
 %STR% "boost software licen[cs]e"
@@ -7865,11 +7965,11 @@ k
 #
 %ENTRY% _TITLE_CDDL_10
 %KEY% "licen[cs]"
-%STR% "common development and distribution licen[cs]e (cddl )?(v|version )1\.?0"
+%STR% "common development and distribution licen[cs]e (cddl )?(v|version )?1\.?0"
 #
 %ENTRY% _TITLE_CDDL_11
 %KEY% "licen[cs]"
-%STR% "common development and distribution licen[cs]e (cddl )?(v|version )1\.?1"
+%STR% "common development and distribution licen[cs]e (cddl )?(v|version )?1\.?1"
 #
 %ENTRY% _TITLE_CDLA_Permissive_10
 %KEY% "licen[cs]"
@@ -8417,6 +8517,10 @@ k
 %KEY% "\<([mn]pl|mozilla|netscape)\>"
 %STR% "mozilla =SOME= end[ -]user software licen[cs]e agreement (v|version )1\.?1"
 #
+%ENTRY% _TITLE_MulanPSL_10
+%KEY% "licen[cs]"
+%STR% "Mulan Permissive Software License (version |v)1"
+#
 %ENTRY% _TITLE_MYSQL_V03
 %KEY% "licen[cs]"
 %STR% "mysql floss licen[cs]e exception version 0\.?3"
@@ -8789,10 +8893,14 @@ k
 %KEY% "licen[cs]"
 %STR% "(licen[cs] agreement for python 2\.?4\.?4|python 2\.?4\.?4 licen[cs]e)"
 #
-%ENTRY% _TITLE_PYTHON2
+%ENTRY% _TITLE_PYTHON_20_1
 %KEY% "licen[cs]"
 %STR% "(licen[cs]e agreement for python 2|python 2 licen[cs]e)"
 #
+%ENTRY% _TITLE_PYTHON_20_2
+%KEY% "licen[cs]"
+%STR% "python software foundation licen[cs]e v?2"
+#
 %ENTRY% _TITLE_QUEST_EULA
 %KEY% "licen[cs]"
 %STR% "quest software inc software licen[cs]e agreement"
@@ -8985,7 +9093,7 @@ k
 %KEY% =NULL=
 %STR% "Universal FOSS Exception (Version |v\.)1.0"
 #
-%ENTRY% _TITLE_VIM
+%ENTRY% _TITLE_Vim
 %KEY% "licen[cs]"
 %STR% "vim licen[cs]e"
 #
@@ -9150,12 +9258,14 @@ k
 %KEY% =NULL=
 %STR% "^begin "
 #
+%ENTRY% _PHR_0BSD
+%KEY% "licen[cs]"
+%STR% "(released|distributed|licen[cs]ed) under (the )?zero clause bsd licen[cs]e"
+#
 %ENTRY% _PHR_AFL
 %KEY% "licen[cs]"
 %STR% "academic free licen[cs]e =SOME= (v|version)"
 #
-
-
 %ENTRY% _PHR_AGPL
 %KEY% "licen[cs]"
 %STR% "affero (general p|p)ublic licen[cs]e"
@@ -9183,9 +9293,12 @@ k
 %KEY% =NULL=
 %STR% "(license|available under) =FEW= AGPL( ?v\.?3|[- ]3\.?0| version 3)"
 #
+# From the regex point of view '( to)?' is unnecessary, however, =FEW= matches
+# up to 30 chars and f.ex. in case 'subject to terms and conditions of the
+# Apache License' Apache license is missed if regex uses 'subject' only.
 %ENTRY% _PHR_Apache_ref0
 %KEY% =NULL=
-%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject|comes|used|software|\<is\>|code|copyright|www|under|corresponding to) =FEW= apache"
+%STR% "(free|distribu|develop|offere?d?|covered|released?|licen[cs]ed?|available|protected|provided|subject( to)?|comes|used|software|\<is\>|code|copyright|www|under|corresponding to) =FEW= apache"
 #
 %ENTRY% _PHR_Apache_ref1
 %KEY% "\<ap(ache|ple)\>"
@@ -9226,7 +9339,7 @@ k
 #
 %ENTRY% _PHR_Apache_20_ref2
 %KEY% "\<ap(ache|ple)\>"
-%STR% "(covered by|(licen[cs]ed|is|released) under) =FEW= apache(( |-)2\.?0| ?v2)"
+%STR% "(covered by|(licen[cs]ed|is|released|provided) under) =FEW= apache(( |-)2\.?0| ?v2)"
 #
 %ENTRY% _PHR_Apache_20_ref3
 %KEY% "licen[cs]"
@@ -9234,7 +9347,7 @@ k
 #
 %ENTRY% _PHR_Apache_20_ref4
 %KEY% "\<ap(ache|ple)\>"
-%STR% "(distribut(able|ed?)|offere?d?|released?|licen[cs]ed?|available|protected|provided) under =SOME= apache licen[cs]e =FEW= (v|version )2\.?0?"
+%STR% "(distribut(able|ed?)|offere?d?|released?|licen[cs]ed?|available|protected|provided) under =SOME= apache (software )?licen[cs]e =FEW= (v|version )2\.?0?"
 #
 %ENTRY% _PHR_Apache_20_ref5
 %KEY% "licen[cs]"
@@ -9244,6 +9357,14 @@ k
 %KEY% "licen[cs]"
 %STR% "license =FEW= osi approved =FEW= apache (software )?license"
 #
+%ENTRY% _PHR_Apache_20_ref7
+%KEY% "licen[cs]"
+%STR% "subject to =FEW= apache (software )?license (v(ersion )?)2"
+#
+%ENTRY% _PHR_Artistic_20
+%KEY% "licen[cs]"
+%STR% "(distribut(able|ed?)|offere?d?|released?|licen[cs]ed?|available|protected|provided) under =FEW= artistic (licen[cs]e 2\.?0|2\.?0 licen[cs]e)"
+#
 %ENTRY% _PHR_ARTISTIC_DESC1
 %KEY% "distribut"
 %STR% "licen[cs]es Perl allows itself to be r?e?-?distributed and/or modified for the other one th(e perl|e) artistic licen[cs]e"
@@ -9260,6 +9381,10 @@ k
 %KEY% "a(gree|ttribut|vailabl|uthor)"
 %STR% "attribution =SOME= (present|provide|display|include)"
 #
+%ENTRY% _PHR_Autoconf_exception_30
+%KEY% "configur"
+%STR% "Autoconf Configure Script Exception version 3\.?0"
+#
 %ENTRY% _PHR_BOOST_V10
 %KEY% "licen[cs]"
 %STR% "boost software licen[cs]e (v|version )1\.?0"
@@ -9322,7 +9447,7 @@ k
 #
 %ENTRY% _PHR_CC0_2
 %KEY% "licen[cs]"
-%STR% "under the terms of a creative commons cc0 1\.?0 licen[cs]e"
+%STR% "under the (terms of a )?creative commons (zero )?cc0 (1\.?0 )?licen[cs]e"
 #
 %ENTRY% _PHR_CC_BY_SA_30
 %KEY% "licen[cs]"
@@ -9340,13 +9465,13 @@ k
 %KEY% "commerc"
 %STR% "(commerciall?y? (and|or|and/or) non[ -]?commerc|non[ -]?commerciall?y? (and|or|and/or) commerc)"
 #
-%ENTRY% _PHR_COPYLEFT_NEXT_030
-%KEY% "licen[cs]"
-%STR% "(r?e?distributed|available|licensed) under copyleft-next( |-)0\.?3\.?0 (licen[cs]e)?"
+%ENTRY% _PHR_copyleft_next_030
+%KEY% "copyleft"
+%STR% "(r?e?distributed?|available|licensed|modify) =FEW= under (the terms of )?copyleft-next( |-| v| version )0\.?3\.?0"
 #
-%ENTRY% _PHR_COPYLEFT_NEXT_031
-%KEY% "licen[cs]"
-%STR% "(r?e?distributed|available|licensed) under copyleft-next( |-)0\.?3\.?1 (licen[cs]e)?"
+%ENTRY% _PHR_copyleft_next_031
+%KEY% "copyleft"
+%STR% "(r?e?distributed?|available|licensed|modify) =FEW= under (the terms of )?copyleft-next( |-| v| version )0\.?3\.?1"
 #
 %ENTRY% _PHR_DOC
 %KEY% "licen[cs]e"
@@ -9468,7 +9593,7 @@ k
 #
 %ENTRY% _PHR_GPL2_ONLY_1
 %KEY% "(\<gnu|free\>)"
-%STR% "version 2 of (the )?gnu general public license"
+%STR% "version 2\.?0? of (the )?gnu general public license"
 #
 %ENTRY% _PHR_GPL2_ONLY_2
 %KEY% "(\<gnu|free\>)"
@@ -9728,7 +9853,7 @@ k
 #
 %ENTRY% _PHR_LGPL2_OR_LATER_2
 %KEY% "(\<gnu|free\>|l?gpl)"
-%STR% "lgpl( |-|v)2\.?0?\+"
+%STR% "lgpl( |-| ?v)2\.?0?\+"
 #
 %ENTRY% _PHR_LGPL2_OR_LATER_3
 %KEY% "(\<gnu|free\>|l?gpl)"
@@ -9796,7 +9921,7 @@ k
 #
 %ENTRY% _PHR_LGPL3_OR_LATER_ref3
 %KEY% =NULL=
-%STR% "lgpl( ?v|-)?3\.?0?-or-later"
+%STR% "lgpl( ?v|-)?3\.?0?(-or-later|\+)"
 #
 %ENTRY% _PHR_LGPL_30plus
 %KEY% "licen[cs]e"
@@ -10033,6 +10158,10 @@ k
 %KEY% "restrict"
 %STR% "government is subject to restrictions as set forth in"
 #
+%ENTRY% _PHR_Vim
+%KEY% "licen[cs]"
+%STR% "(r?e?distribut|modif(y|ied)|licen[cs]ed?|deliver|use|cop(y|ied)) =SOME= under (the terms of )?(the )?Vim licen[cs]e"
+#
 %ENTRY% _PHR_WTFPL
 %KEY% "licen[cs]"
 %STR% "licen[cs]e =FEW= \<WTFPL\>"
@@ -10116,7 +10245,7 @@ k
 
 %ENTRY% _SPDX_AGPL_10plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(AGPL[- v]1\.?0\+|(.{1,32} (AND|OR)){1,4} AGPL[- v]1\.?0\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(AGPL[- v]1\.?0\+|(.{1,32} (AND|OR)){1,4} AGPL[- v]1\.?0\+)"
 
 %ENTRY% _SPDX_AGPL_30
 %KEY% "licen[cs]e"
@@ -10128,7 +10257,7 @@ k
 
 %ENTRY% _SPDX_AGPL_30plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(AGPL[- v]3\.?0\+|(.{1,32} (AND|OR)){1,4} AGPL[- v]3\.?0\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(AGPL[- v]3\.?0\+|(.{1,32} (AND|OR)){1,4} AGPL[- v]3\.?0\+)"
 
 %ENTRY% _SPDX_Aladdin
 %KEY% "licen[cs]e"
@@ -10230,6 +10359,14 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BitTorrent[- ]1\.?1|(.{1,32} (AND|OR)){1,4} BitTorrent[- ]1\.?1)"
 
+%ENTRY% _SPDX_blessing
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](blessing|(.{1,32} (AND|OR)){1,4} blessing)"
+
+%ENTRY% _SPDX_BlueOak_100
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BlueOak[- ]1\.?0\.?0|(.{1,32} (AND|OR)){1,4} BlueOak[- ]1\.?0\.?0)"
+
 %ENTRY% _SPDX_BSL_10
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} BSL[- ]1\.?0)"
@@ -10262,6 +10399,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSD-2-Clause-Patent|(.{1,32} (AND|OR)){1,4} BSD-2-Clause-Patent)"
 
+%ENTRY% _SPDX_BSD_2_Clause_Views
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSD-2-Clause-Views|(.{1,32} (AND|OR)){1,4} BSD-2-Clause-Views)"
+
 %ENTRY% _SPDX_BSD_3_Clause
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSD-3-Clause|(.{1,32} (AND|OR)){1,4} BSD-3-Clause)"
@@ -10282,6 +10423,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSD-3-Clause-No-Nuclear-Warranty|(.{1,32} (AND|OR)){1,4} BSD-3-Clause-No-Nuclear-Warranty)"
 
+%ENTRY% _SPDX_BSD_3_Clause_Open_MPI
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSD-3-Clause-Open-MPI|(.{1,32} (AND|OR)){1,4} BSD-3-Clause-Open-MPI)"
+
 %ENTRY% _SPDX_BSD_4_Clause
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](BSD-4-Clause|(.{1,32} (AND|OR)){1,4} BSD-4-Clause)"
@@ -10310,89 +10455,25 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](bzip2[- ]1\.?0\.?6|(.{1,32} (AND|OR)){1,4} bzip2[- ]1\.?0\.?6)"
 
-%ENTRY% _SPDX_Caldera
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Caldera|(.{1,32} (AND|OR)){1,4} Caldera)"
-
-%ENTRY% _SPDX_CDLA_Permissive_10
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDLA-Permissive-1\.?0|(.{1,32} (AND|OR)){1,4} CDLA-Permissive-1\.?0)"
-
-%ENTRY% _SPDX_CDLA_Sharing_10
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDLA-Sharing-1\.?0|(.{1,32} (AND|OR)){1,4} CDLA-Sharing-1\.?0)"
-
-%ENTRY% _SPDX_CECILL_10
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-1\.?0|(.{1,32} (AND|OR)){1,4} CECILL-1\.?0)"
-
-%ENTRY% _SPDX_CECILL_11
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-1\.?1|(.{1,32} (AND|OR)){1,4} CECILL-1\.?1)"
-
-%ENTRY% _SPDX_CECILL_20
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL[- ]2\.?0|(.{1,32} (AND|OR)){1,4} CECILL[- ]2\.?0)"
-
-%ENTRY% _SPDX_CECILL_21
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL[- ]2\.?1|(.{1,32} (AND|OR)){1,4} CECILL[- ]2\.?1)"
-
-%ENTRY% _SPDX_CECILL_B
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-B|(.{1,32} (AND|OR)){1,4} CECILL-B)"
-
-%ENTRY% _SPDX_CECILL_C
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-C|(.{1,32} (AND|OR)){1,4} CECILL-C)"
-
-%ENTRY% _SPDX_ClArtistic
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](ClArtistic|(.{1,32} (AND|OR)){1,4} ClArtistic)"
-
-%ENTRY% _SPDX_MIT_CMU
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](MIT-CMU|(.{1,32} (AND|OR)){1,4} MIT-CMU)"
-
-%ENTRY% _SPDX_CNRI_Jython
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CNRI-Jython|(.{1,32} (AND|OR)){1,4} CNRI-Jython)"
-
-%ENTRY% _SPDX_CNRI_Python
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CNRI-Python|(.{1,32} (AND|OR)){1,4} CNRI-Python)"
-
-%ENTRY% _SPDX_CNRI_Python_GPL_Compatible
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CNRI-Python-GPL-Compatible|(.{1,32} (AND|OR)){1,4} CNRI-Python-GPL-Compatible)"
-
-%ENTRY% _SPDX_CPOL_102
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CPOL[- ]1\.?02|(.{1,32} (AND|OR)){1,4} CPOL[- ]1\.?02)"
-
-%ENTRY% _SPDX_CDDL_10
-%KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDDL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} CDDL[- ]1\.?0)"
-
-%ENTRY% _SPDX_CDDL_11
+%ENTRY% _SPDX_CAL_10_Combined_Work_Exception
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDDL[- ]1\.?1|(.{1,32} (AND|OR)){1,4} CDDL[- ]1\.?1)"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CAL-1\.?0-Combined-Work-Exception|(.{1,32} (AND|OR)){1,4} CAL-1\.?0-Combined-Work-Exception)"
 
-%ENTRY% _SPDX_CPAL_10
+%ENTRY% _SPDX_CAL_10
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CPAL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} CPAL[- ]1\.?0)"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CAL-1\.?0|(.{1,32} (AND|OR)){1,4} CAL-1\.?0)"
 
-%ENTRY% _SPDX_CPL_10
+%ENTRY% _SPDX_Caldera
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} CPL[- ]1\.?0)"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Caldera|(.{1,32} (AND|OR)){1,4} Caldera)"
 
 %ENTRY% _SPDX_CATOSL_11
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CATOSL[- ]1\.?1|(.{1,32} (AND|OR)){1,4} CATOSL[- ]1\.?1)"
 
-%ENTRY% _SPDX_Condor_11
+%ENTRY% _SPDX_CC0_10
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Condor[- ]1\.?1|(.{1,32} (AND|OR)){1,4} Condor[- ]1\.?1)"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC0-1\.?0|(.{1,32} (AND|OR)){1,4} CC0-1\.?0)"
 
 %ENTRY% _SPDX_CC_BY_10
 %KEY% "licen[cs]e"
@@ -10410,6 +10491,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-3\.?0|(.{1,32} (AND|OR)){1,4} CC-BY-3\.?0)"
 
+%ENTRY% _SPDX_CC_BY_30_AT
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-3\.?0-AT|(.{1,32} (AND|OR)){1,4} CC-BY-3\.?0-AT)"
+
 %ENTRY% _SPDX_CC_BY_40
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-4\.?0|(.{1,32} (AND|OR)){1,4} CC-BY-4\.?0)"
@@ -10470,6 +10555,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-NC-ND-3\.?0|(.{1,32} (AND|OR)){1,4} CC-BY-NC-ND-3\.?0)"
 
+%ENTRY% _SPDX_CC_BY_NC_ND_30_IGO
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-NC-ND-3\.?0-IGO|(.{1,32} (AND|OR)){1,4} CC-BY-NC-ND-3\.?0-IGO)"
+
 %ENTRY% _SPDX_CC_BY_NC_ND_40
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-NC-ND-4\.?0|(.{1,32} (AND|OR)){1,4} CC-BY-NC-ND-4\.?0)"
@@ -10510,13 +10599,101 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-SA-3\.?0|(.{1,32} (AND|OR)){1,4} CC-BY-SA-3\.?0)"
 
+%ENTRY% _SPDX_CC_BY_SA_30_AT
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-SA-3\.?0-AT|(.{1,32} (AND|OR)){1,4} CC-BY-SA-3\.?0-AT)"
+
 %ENTRY% _SPDX_CC_BY_SA_40
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-BY-SA-4\.?0|(.{1,32} (AND|OR)){1,4} CC-BY-SA-4\.?0)"
 
-%ENTRY% _SPDX_CC0_10
+%ENTRY% _SPDX_CC_PDDC
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC0-1\.?0|(.{1,32} (AND|OR)){1,4} CC0-1\.?0)"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CC-PDDC|(.{1,32} (AND|OR)){1,4} CC-PDDC)"
+
+%ENTRY% _SPDX_CDDL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDDL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} CDDL[- ]1\.?0)"
+
+%ENTRY% _SPDX_CDDL_11
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDDL[- ]1\.?1|(.{1,32} (AND|OR)){1,4} CDDL[- ]1\.?1)"
+
+%ENTRY% _SPDX_CDLA_Permissive_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDLA-Permissive-1\.?0|(.{1,32} (AND|OR)){1,4} CDLA-Permissive-1\.?0)"
+
+%ENTRY% _SPDX_CDLA_Sharing_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CDLA-Sharing-1\.?0|(.{1,32} (AND|OR)){1,4} CDLA-Sharing-1\.?0)"
+
+%ENTRY% _SPDX_CECILL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-1\.?0|(.{1,32} (AND|OR)){1,4} CECILL-1\.?0)"
+
+%ENTRY% _SPDX_CECILL_11
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-1\.?1|(.{1,32} (AND|OR)){1,4} CECILL-1\.?1)"
+
+%ENTRY% _SPDX_CECILL_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL[- ]2\.?0|(.{1,32} (AND|OR)){1,4} CECILL[- ]2\.?0)"
+
+%ENTRY% _SPDX_CECILL_21
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL[- ]2\.?1|(.{1,32} (AND|OR)){1,4} CECILL[- ]2\.?1)"
+
+%ENTRY% _SPDX_CECILL_B
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-B|(.{1,32} (AND|OR)){1,4} CECILL-B)"
+
+%ENTRY% _SPDX_CECILL_C
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CECILL-C|(.{1,32} (AND|OR)){1,4} CECILL-C)"
+
+%ENTRY% _SPDX_CERN_OHL_11
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CERN-OHL[- ]1\.?1|(.{1,32} (AND|OR)){1,4} CERN-OHL[- ]1\.?1)"
+
+%ENTRY% _SPDX_CERN_OHL_12
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CERN-OHL[- ]1\.?2|(.{1,32} (AND|OR)){1,4} CERN-OHL[- ]1\.?2)"
+
+%ENTRY% _SPDX_CERN_OHL_P_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CERN-OHL-P[- ]2\.?0|(.{1,32} (AND|OR)){1,4} CERN-OHL-P[- ]2\.?0)"
+
+%ENTRY% _SPDX_CERN_OHL_S_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CERN-OHL-S[- ]2\.?0|(.{1,32} (AND|OR)){1,4} CERN-OHL-S[- ]2\.?0)"
+
+%ENTRY% _SPDX_CERN_OHL_W_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CERN-OHL-W[- ]2\.?0|(.{1,32} (AND|OR)){1,4} CERN-OHL-W[- ]2\.?0)"
+
+%ENTRY% _SPDX_ClArtistic
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](ClArtistic|(.{1,32} (AND|OR)){1,4} ClArtistic)"
+
+%ENTRY% _SPDX_MIT_CMU
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](MIT-CMU|(.{1,32} (AND|OR)){1,4} MIT-CMU)"
+
+%ENTRY% _SPDX_CNRI_Jython
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CNRI-Jython|(.{1,32} (AND|OR)){1,4} CNRI-Jython)"
+
+%ENTRY% _SPDX_CNRI_Python
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CNRI-Python|(.{1,32} (AND|OR)){1,4} CNRI-Python)"
+
+%ENTRY% _SPDX_CNRI_Python_GPL_Compatible
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CNRI-Python-GPL-Compatible|(.{1,32} (AND|OR)){1,4} CNRI-Python-GPL-Compatible)"
+
+%ENTRY% _SPDX_Condor_11
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Condor[- ]1\.?1|(.{1,32} (AND|OR)){1,4} Condor[- ]1\.?1)"
 
 %ENTRY% _SPDX_copyleft_next_030
 %KEY% "licen[cs]e"
@@ -10526,6 +10703,18 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](copyleft-next-0\.?3\.?1|(.{1,32} (AND|OR)){1,4} copyleft-next-0\.?3\.?1)"
 
+%ENTRY% _SPDX_CPAL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CPAL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} CPAL[- ]1\.?0)"
+
+%ENTRY% _SPDX_CPL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} CPL[- ]1\.?0)"
+
+%ENTRY% _SPDX_CPOL_102
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](CPOL[- ]1\.?02|(.{1,32} (AND|OR)){1,4} CPOL[- ]1\.?02)"
+
 %ENTRY% _SPDX_Crossword
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Crossword|(.{1,32} (AND|OR)){1,4} Crossword)"
@@ -10556,7 +10745,7 @@ k
 
 %ENTRY% _SPDX_DOC
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(DOC\>|(.{1,32} (AND|OR)){1,4} DOC\>)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(DOC\>|(.{1,32} (AND|OR)){1,4} DOC\>)"
 
 %ENTRY% _SPDX_Dotseqn
 %KEY% "licen[cs]e"
@@ -10606,10 +10795,18 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Entessa|(.{1,32} (AND|OR)){1,4} Entessa)"
 
+%ENTRY% _SPDX_EPICS
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](EPICS|(.{1,32} (AND|OR)){1,4} EPICS)"
+
 %ENTRY% _SPDX_ErlPL_11
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](ErlPL[- ]1\.?1|(.{1,32} (AND|OR)){1,4} ErlPL[- ]1\.?1)"
 
+%ENTRY% _SPDX_etalab_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](etalab[- ]2\.?0|(.{1,32} (AND|OR)){1,4} etalab[- ]2\.?0)"
+
 %ENTRY% _SPDX_EUDatagrid
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](EUDatagrid|(.{1,32} (AND|OR)){1,4} EUDatagrid)"
@@ -10632,7 +10829,7 @@ k
 
 %ENTRY% _SPDX_Fair
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(Fair[[:space:]|\"]|(.{1,32} (AND|OR)){1,4} Fair[[:space:]|\"])"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(Fair[[:space:]|\"]|(.{1,32} (AND|OR)){1,4} Fair[[:space:]|\"])"
 
 %ENTRY% _SPDX_Frameworx_10
 %KEY% "licen[cs]e"
@@ -10672,7 +10869,23 @@ k
 
 %ENTRY% _SPDX_GFDL_11plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(GFDL[- v]1\.?1\+|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?1\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(GFDL[- v]1\.?1\+|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?1\+)"
+
+%ENTRY% _SPDX_GFDL_11_invariants_only
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?1-invariants-only|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?1-invariants-only)"
+
+%ENTRY% _SPDX_GFDL_11_invariants_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?1-invariants.or.later|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?1-invariants.or.later)"
+
+%ENTRY% _SPDX_GFDL_11_no_invariants_only
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?1-no-invariants-only|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?1-no-invariants-only)"
+
+%ENTRY% _SPDX_GFDL_11_no_invariants_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?1-no-invariants.or.later|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?1-no-invariants.or.later)"
 
 %ENTRY% _SPDX_GFDL_12
 %KEY% "licen[cs]e"
@@ -10684,7 +10897,23 @@ k
 
 %ENTRY% _SPDX_GFDL_12plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(GFDL[- v]1\.?2\+|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?2\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(GFDL[- v]1\.?2\+|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?2\+)"
+
+%ENTRY% _SPDX_GFDL_12_invariants_only
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?2-invariants-only|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?2-invariants-only)"
+
+%ENTRY% _SPDX_GFDL_12_invariants_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?2-invariants.or.later|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?2-invariants.or.later)"
+
+%ENTRY% _SPDX_GFDL_12_no_invariants_only
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?2-no-invariants-only|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?2-no-invariants-only)"
+
+%ENTRY% _SPDX_GFDL_12_no_invariants_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?2-no-invariants.or.later|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?2-no-invariants.or.later)"
 
 %ENTRY% _SPDX_GFDL_13
 %KEY% "licen[cs]e"
@@ -10696,7 +10925,23 @@ k
 
 %ENTRY% _SPDX_GFDL_13plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(GFDL[- v]1\.?3\+|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?3\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(GFDL[- v]1\.?3\+|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?3\+)"
+
+%ENTRY% _SPDX_GFDL_13_invariants_only
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?3-invariants-only|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?3-invariants-only)"
+
+%ENTRY% _SPDX_GFDL_13_invariants_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?3-invariants.or.later|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?3-invariants.or.later)"
+
+%ENTRY% _SPDX_GFDL_13_no_invariants_only
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?3-no-invariants-only|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?3-no-invariants-only)"
+
+%ENTRY% _SPDX_GFDL_13_no_invariants_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GFDL[- v]1\.?3-no-invariants.or.later|(.{1,32} (AND|OR)){1,4} GFDL[- v]1\.?3-no-invariants.or.later)"
 
 %ENTRY% _SPDX_GL2PS
 %KEY% "licen[cs]e"
@@ -10706,25 +10951,33 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Glulxe|(.{1,32} (AND|OR)){1,4} Glulxe)"
 
+%ENTRY% _SPDX_GLWTPL
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](GLWTPL|(.{1,32} (AND|OR)){1,4} GLWTPL)"
+
 %ENTRY% _SPDX_GPL_10
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[: \[]+(GPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} GPL[- ]1\.?0)"
+%STR% "licen[cs]e(id|[- ]identifier)?[: \[\(]+(GPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} GPL[- ]1\.?0)"
 
 %ENTRY% _SPDX_GPL_10plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(GPL[- ]1\.?0\+|(.{1,32} (AND|OR)){1,4} GPL[- ]1\.?0\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(GPL[- ]1\.?0\+|(.{1,32} (AND|OR)){1,4} GPL[- ]1\.?0\+)"
 
 %ENTRY% _SPDX_GPL_20
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[: \[]+(GPL[- ]2\.?0|(.{1,32} (AND|OR)){1,4} GPL[- ]2\.?0)"
+%STR% "licen[cs]e(id|[- ]identifier)?[: \[\(]+(GPL[- ]2\.?0|(.{1,32} (AND|OR)){1,4} GPL[- ]2\.?0)"
+
+%ENTRY% _SPDX_GPL_20_or_later
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier)?[: \[\(]+(GPL[- ]2\.?0-or-later|(.{1,32} (AND|OR)){1,4} GPL[- ]2\.?0-or-later)"
 
 %ENTRY% _SPDX_GPL_30
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[: \[]+GPL[- ]3\.?0)"
+%STR% "licen[cs]e(id|[- ]identifier)?[: \[\(]+GPL[- ]3\.?0)"
 
 %ENTRY% _SPDX_GPL_30plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(GPL[- ]3\.?0\+|(.{1,32} (AND|OR)){1,4} GPL[- ]3\.?0\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(GPL[- ]3\.?0\+|(.{1,32} (AND|OR)){1,4} GPL[- ]3\.?0\+)"
 
 %ENTRY% _SPDX_LGPL_20
 %KEY% "licen[cs]e"
@@ -10740,7 +10993,7 @@ k
 
 %ENTRY% _SPDX_LGPL_30plus
 %KEY% "licen[cs]e"
-%STR% "licen[cs]e(id|[- ]identifier)?[\[\":[:space:]]+(LGPL[- ]3\.?0\+|(.{1,32} (AND|OR)){1,4} LGPL[- ]3\.?0\+)"
+%STR% "licen[cs]e(id|[- ]identifier)?[\[\"\(:[:space:]]+(LGPL[- ]3\.?0\+|(.{1,32} (AND|OR)){1,4} LGPL[- ]3\.?0\+)"
 
 %ENTRY% _SPDX_gnuplot
 %KEY% "licen[cs]e"
@@ -10754,10 +11007,18 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](HaskellReport|(.{1,32} (AND|OR)){1,4} HaskellReport)"
 
+%ENTRY% _SPDX_Hippocratic_21
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Hippocratic[- ]2\.?1|(.{1,32} (AND|OR)){1,4} Hippocratic[- ]2\.?1)"
+
 %ENTRY% _SPDX_HPND
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](HPND|(.{1,32} (AND|OR)){1,4} HPND)"
 
+%ENTRY% _SPDX_HPND_sell_variant
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](HPND-sell-variant|(.{1,32} (AND|OR)){1,4} HPND-sell-variant)"
+
 %ENTRY% _SPDX_IBM_pibs
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](IBM-pibs|(.{1,32} (AND|OR)){1,4} IBM-pibs)"
@@ -10814,6 +11075,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](JasPer[- ]2\.?0|(.{1,32} (AND|OR)){1,4} JasPer[- ]2\.?0)"
 
+%ENTRY% _SPDX_JPNIC
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](\<JPNIC\>|(.{1,32} (AND|OR)){1,4} \<JPNIC\>)"
+
 %ENTRY% _SPDX_JSON
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](\<JSON\>|(.{1,32} (AND|OR)){1,4} \<JSON\>)"
@@ -10858,6 +11123,14 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Libpng|(.{1,32} (AND|OR)){1,4} Libpng)"
 
+%ENTRY% _SPDX_libpng_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](libpng[- ]2\.?0|(.{1,32} (AND|OR)){1,4} libpng[- ]2\.?0)"
+
+%ENTRY% _SPDX_libselinux_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](libselinux[- ]1\.?0|(.{1,32} (AND|OR)){1,4} libselinux[- ]1\.?0)"
+
 %ENTRY% _SPDX_libtiff
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](libtiff|(.{1,32} (AND|OR)){1,4} libtiff)"
@@ -10962,6 +11235,14 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](MTLL|(.{1,32} (AND|OR)){1,4} MTLL)"
 
+%ENTRY% _SPDX_MulanPSL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](MulanPSL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} MulanPSL[- ]1\.?0)"
+
+%ENTRY% _SPDX_MulanPSL_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](MulanPSL[- ]2\.?0|(.{1,32} (AND|OR)){1,4} MulanPSL[- ]2\.?0)"
+
 %ENTRY% _SPDX_Multics
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Multics|(.{1,32} (AND|OR)){1,4} Multics)"
@@ -10982,6 +11263,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NBPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} NBPL[- ]1\.?0)"
 
+%ENTRY% _SPDX_NCGL_UK_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NCGL-UK[- ]2\.?0|(.{1,32} (AND|OR)){1,4} NCGL-UK[- ]2\.?0)"
+
 %ENTRY% _SPDX_NCSA
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NCSA|(.{1,32} (AND|OR)){1,4} NCSA)"
@@ -11002,6 +11287,14 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NGPL|(.{1,32} (AND|OR)){1,4} NGPL)"
 
+%ENTRY% _SPDX_NIST_PD
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NIST-PD|(.{1,32} (AND|OR)){1,4} NIST-PD)"
+
+%ENTRY% _SPDX_NIST_PD_fallback
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NIST-PD-fallback|(.{1,32} (AND|OR)){1,4} NIST-PD-fallback)"
+
 %ENTRY% _SPDX_NLOD_10
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NLOD[- ]1\.?0|(.{1,32} (AND|OR)){1,4} NLOD[- ]1\.?0)"
@@ -11038,6 +11331,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](NRL|(.{1,32} (AND|OR)){1,4} NRL)"
 
+%ENTRY% _SPDX_NTP_0
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](\<NTP-0\>|(.{1,32} (AND|OR)){1,4} \<NTP-0\>)"
+
 %ENTRY% _SPDX_NTP
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](\<NTP\>|(.{1,32} (AND|OR)){1,4} \<NTP\>)"
@@ -11046,6 +11343,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Nunit|(.{1,32} (AND|OR)){1,4} Nunit)"
 
+%ENTRY% _SPDX_O_UDA_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](O-UDA[- ]1\.?0|(.{1,32} (AND|OR)){1,4} O-UDA[- ]1\.?0)"
+
 %ENTRY% _SPDX_OCLC_20
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OCLC[- ]2\.?0|(.{1,32} (AND|OR)){1,4} OCLC[- ]2\.?0)"
@@ -11062,14 +11363,38 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](ODC-By-1\.?0|(.{1,32} (AND|OR)){1,4} ODC-By-1\.?0)"
 
+%ENTRY% _SPDX_OFL_10_no_RFN
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OFL-1\.?0-no-RFN|(.{1,32} (AND|OR)){1,4} OFL-1\.?0-no-RFN)"
+
+%ENTRY% _SPDX_OFL_10_RFN
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OFL-1\.?0-RFN|(.{1,32} (AND|OR)){1,4} OFL-1\.?0-RFN)"
+
 %ENTRY% _SPDX_OFL_10
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OFL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} OFL[- ]1\.?0)"
 
+%ENTRY% _SPDX_OFL_11_no_RFN
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OFL-1\.?1-no-RFN|(.{1,32} (AND|OR)){1,4} OFL-1\.?1-no-RFN)"
+
+%ENTRY% _SPDX_OFL_11_RFN
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OFL-1\.?1-RFN|(.{1,32} (AND|OR)){1,4} OFL-1\.?1-RFN)"
+
 %ENTRY% _SPDX_OFL_11
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OFL[- ]1\.?1|(.{1,32} (AND|OR)){1,4} OFL[- ]1\.?1)"
 
+%ENTRY% _SPDX_OGC_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OGC[ -]1\.?0|(.{1,32} (AND|OR)){1,4} OGC[ -]1\.?0)"
+
+%ENTRY% _SPDX_OGL_Canada_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OGL-Canada-2\.?0|(.{1,32} (AND|OR)){1,4} OGL-Canada-2\.?0)"
+
 %ENTRY% _SPDX_OGL_UK_10
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OGL-UK-1\.?0|(.{1,32} (AND|OR)){1,4} OGL-UK-1\.?0)"
@@ -11188,6 +11513,14 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](OSET-PL-2\.?1|(.{1,32} (AND|OR)){1,4} OSET-PL-2\.?1)"
 
+%ENTRY% _SPDX_Parity_600
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Parity[- ]6\.?0\.?0|(.{1,32} (AND|OR)){1,4} Parity[- ]6\.?0\.?0)"
+
+%ENTRY% _SPDX_Parity_700
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Parity[- ]7\.?0\.?0|(.{1,32} (AND|OR)){1,4} Parity[- ]7\.?0\.?0)"
+
 %ENTRY% _SPDX_PDDL_10
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](PDDL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} PDDL[- ]1\.?0)"
@@ -11204,10 +11537,22 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Plexus|(.{1,32} (AND|OR)){1,4} Plexus)"
 
+%ENTRY% _SPDX_PolyForm_Noncommercial_100
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](PolyForm-Noncommercial[- ]1\.?0\.?0|(.{1,32} (AND|OR)){1,4} PolyForm-Noncommercial[- ]1\.?0\.?0)"
+
+%ENTRY% _SPDX_PolyForm_Small_Business_100
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](PolyForm-Small-Business[- ]1\.?0\.?0|(.{1,32} (AND|OR)){1,4} PolyForm-Small-Business[- ]1\.?0\.?0)"
+
 %ENTRY% _SPDX_PostgreSQL
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](PostgreSQL|(.{1,32} (AND|OR)){1,4} PostgreSQL)"
 
+%ENTRY% _SPDX_PSF_20
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](PSF[- ]2\.?0|(.{1,32} (AND|OR)){1,4} PSF[- ]2\.?0)"
+
 %ENTRY% _SPDX_psfrag
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](psfrag|(.{1,32} (AND|OR)){1,4} psfrag)"
@@ -11280,6 +11625,14 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Sendmail-8\.?23|(.{1,32} (AND|OR)){1,4} Sendmail-8\.?23)"
 
+%ENTRY% _SPDX_SHL_051
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](\<SHL-0\.?51|(.{1,32} (AND|OR)){1,4} \<SHL-0\.?51)"
+
+%ENTRY% _SPDX_SHL_05
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](\<SHL-0\.?5|(.{1,32} (AND|OR)){1,4} \<SHL-0\.?5)"
+
 %ENTRY% _SPDX_SGI_B_10
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](SGI-B-1\.?0|(.{1,32} (AND|OR)){1,4} SGI-B-1\.?0)"
@@ -11340,10 +11693,26 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](SPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} SPL[- ]1\.?0)"
 
+%ENTRY% _SPDX_SSH_OpenSSH
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](SSH-OpenSSH|(.{1,32} (AND|OR)){1,4} SSH-OpenSSH)"
+
+%ENTRY% _SPDX_SSH_short
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](SSH-short|(.{1,32} (AND|OR)){1,4} SSH-short)"
+
+%ENTRY% _SPDX_SSPL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](SSPL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} SSPL[- ]1\.?0)"
+
 %ENTRY% _SPDX_SWL
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](SWL|(.{1,32} (AND|OR)){1,4} SWL)"
 
+%ENTRY% _SPDX_TAPR_OHL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](TAPR-OHL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} TAPR-OHL[- ]1\.?0)"
+
 %ENTRY% _SPDX_TCL
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](TCL|(.{1,32} (AND|OR)){1,4} TCL)"
@@ -11372,6 +11741,10 @@ k
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](TU-Berlin-2\.?0|(.{1,32} (AND|OR)){1,4} TU-Berlin-2\.?0)"
 
+%ENTRY% _SPDX_UCL_10
+%KEY% "licen[cs]e"
+%STR% "licen[cs]e(id|[- ]identifier| name)?[: ](UCL[- ]1\.?0|(.{1,32} (AND|OR)){1,4} UCL[- ]1\.?0)"
+
 %ENTRY% _SPDX_Unicode_DFS_2015
 %KEY% "licen[cs]e"
 %STR% "licen[cs]e(id|[- ]identifier| name)?[: ](Unicode-DFS-2015|(.{1,32} (AND|OR)){1,4} Unicode-DFS-2015)"
@@ -11722,11 +12095,11 @@ k
 # [including the GNU Public Licence.]
 %ENTRY% _TEXT_NOT_GPL
 %KEY% "(gnu|gpl)"
-%STR% "not\> =SOME= (\<gnu\>|\<gpl\>)"
+%STR% "not\> =SOME= (gnu (general|public)|\<gpl\>)"
 #
 %ENTRY% _TEXT_NOT_GPL2
 %KEY% "(gnu|gpl)"
-%STR% "not\> =SOME= cop(y|ied) =SOME= another =SOME= (\<gnu\>|\<gpl\>)"
+%STR% "not\> =SOME= cop(y|ied) =SOME= another =SOME= (gnu (general|public)|\<gpl\>)"
 #
 %ENTRY% _TEXT_GPLV1
 %KEY% "(\<gnu|free\>|l?gpl)"
@@ -12079,6 +12452,10 @@ k
 %KEY% "licen[cs]"
 %STR% "w?w?w?\.?opensource\.?org/licen[cs]es?/artistic"
 #
+%ENTRY% _URL_BlueOak_100
+%KEY% "licen[cs]"
+%STR% "blueoakcouncil\.?org/licen[cs]e/1\.?0\.?0"
+#
 %ENTRY% _URL_BOOST_10
 %KEY% "licen[cs]"
 %STR% "w?w?w?\.?boost\.?org/licen[cs]e[_-]?1_?0"
@@ -12223,7 +12600,7 @@ k
 %KEY% "public"
 %STR% "creativecommons\.?org/publicdomain/zero/1\.?0"
 #
-%ENTRY% _URL_CC_PD
+%ENTRY% _URL_CC_PDDC
 %KEY% "public"
 %STR% "creativecommons\.?org/licenses/publicdomain"
 #
@@ -12305,7 +12682,7 @@ k
 #
 %ENTRY% _URL_ISC
 %KEY% =NULL=
-%STR% "http://isc\?.org/copyright"
+%STR% "isc\.?org/copyright"
 #
 %ENTRY% _URL_IMAGEMAGICK
 %KEY% "licen[cs]"
@@ -12359,6 +12736,10 @@ k
 %KEY% "\<([mn]pl|mozilla|netscape)\>"
 %STR% "w?w?w?\.?mozilla\.?org/[MN]PL/MPL-2\.?0"
 #
+%ENTRY% _URL_MulanPSL
+%KEY% "licen[cs]"
+%STR% "license\.?coscl\.?org\.?cn/MulanPSL"
+#
 %ENTRY% _URL_mysql_floss_exception
 %KEY% "licen[cs]"
 %STR% "w?w?w?\.?mysql\.?com/about/legal/licensing/foss-exception"
@@ -12685,7 +13066,7 @@ k
 #
 %ENTRY% _MIN2_GPL
 %KEY% "so(ftware|urce)"
-%STR% "(GPL src-instance|[^l]gpl =FEW= \<source\>|\<source\> =FEW= [^l]gpl)"
+%STR% "(GPL src-instance|\<gpl\> =FEW= \<source\>|\<source\> =FEW= \<gpl\>)"
 #
 %ENTRY% _MIN2_HP
 %KEY% "so(ftware|urce)"
diff --git a/src/nomos/agent/parse.c b/src/nomos/agent/parse.c
index ed4471ec2..0f6cace05 100755
--- a/src/nomos/agent/parse.c
+++ b/src/nomos/agent/parse.c
@@ -612,10 +612,6 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     else if (INFILE(_CR_SLEEPYCAT) || INFILE(_LT_SLEEPYCAT_1)) {
       MEDINTEREST(lDebug ? "Sleepycat(1)" : "Sleepycat");
     }
-    else if (INFILE(_TITLE_PHP202)) {
-      INTERESTING(lDebug ? "PHP(v2.02#2)" : "PHP-2.02");
-      lmem[_mPHP] = 1;
-    }
     else if (INFILE(_TITLE_ZEND_V20)) {
       INTERESTING("Zend-2.0");
     }
@@ -633,6 +629,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
         else if (INFILE(_LT_BSD_CLAUSE_4) && INFILE(_LT_BSD_CLAUSE_CLEAR)) {
           INTERESTING("BSD-3-Clause-Clear");
         }
+        else if (INFILE(_LT_BSD_CLAUSE_4) && INFILE(_LT_BSD_CLAUSE_OPEN_MPI)) {
+          INTERESTING("BSD-3-Clause-Open-MPI");
+        }
         else if (INFILE(_LT_BSD_CLAUSE_4) && HASTEXT(_KW_severability, REG_EXTENDED)) {
           INTERESTING("BSD-3-Clause-Severability");
         }
@@ -652,7 +651,12 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
           INTERESTING("BSD-3-Clause-Attribution");
         }
         else if (!lmem[_mAPACHE10] && !lmem[_mAPACHE11] && INFILE(_LT_BSD_CLAUSE_4)) {
-          INTERESTING("BSD-3-Clause");
+          if (INFILE(_LT_DARPA_COUGAAR_2)) {
+            INTERESTING("DARPA-Cougaar");
+          }
+          else {
+            INTERESTING("BSD-3-Clause");
+          }
         }
         else if (INFILE(_LT_SSLEAY)) {
           INTERESTING("SSLeay");
@@ -706,6 +710,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
       else if (INFILE(_LT_BSD_CLAUSE_1_DISCLAIMER)) {
         INTERESTING("BSD-1-Clause");
       }
+      else if (INFILE(_LT_Spencer_99) && INFILE(_CR_Spencer)) {
+        INTERESTING("Spencer-99");
+      }
       else if (!lmem[_fZPL]) {
         INTERESTING(lDebug ? "BSD-style(1)" : "BSD-style");
       }
@@ -762,6 +769,10 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
       INTERESTING("WordNet-3.0");
       lmem[_mMIT] = 1;
     }
+    else if (INFILE(_LT_HPND_1) && INFILE(_LT_HPND_2)) {
+      INTERESTING("HPND");
+      lmem[_mMIT] = 1;
+    }
     else if (INFILE(_LT_NOT_ADVERTISING)) {
       INTERESTING(lDebug ? "BSD-style(2)" : "BSD-style");
     }
@@ -783,6 +794,12 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     else if (INFILE(_TITLE_OZPLB_10)) {
       INTERESTING("OZPLB-1.0");
     }
+    /*
+     * JPNIC
+     */
+    else if (HASTEXT(_TEXT_JPNIC, 0) && INFILE(_LT_JPNIC)) {
+      INTERESTING("JPNIC");
+    }
     else if (NOT_INFILE(_CR_XFREE86) && NOT_INFILE(_TITLE_NCSA) && NOT_INFILE(_TITLE_INNERNET200)) {
       INTERESTING(lDebug ? "BSD-style(3)" : "BSD-style");
     }
@@ -919,6 +936,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (INFILE(_SPDX_BSD_3_Clause_LBNL)) {
     INTERESTING("BSD-3-Clause-LBNL");
   }
+  else if (INFILE(_SPDX_BSD_3_Clause_Open_MPI)) {
+    INTERESTING("BSD-3-Clause-Open-MPI");
+  }
   else if (INFILE(_SPDX_BSD_3_Clause)) {
     INTERESTING("BSD-3-Clause");
   }
@@ -934,6 +954,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (INFILE(_SPDX_BSD_2_Clause_Patent)) {
     INTERESTING("BSD-2-Clause-Patent");
   }
+  else if (INFILE(_SPDX_BSD_2_Clause_Views)) {
+    INTERESTING("BSD-2-Clause-Views");
+  }
   else if (INFILE(_SPDX_BSD_2_Clause_1)) {
     INTERESTING("BSD-2-Clause");
   }
@@ -972,6 +995,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (INFILE(_SPDX_BSD_1_Clause)) {
     INTERESTING("BSD-1-Clause");
   }
+  else if (INFILE(_PHR_0BSD)) {
+    INTERESTING("0BSD");
+  }
   else if (INFILE(_LT_BSDref1)) {
     INTERESTING(lDebug ? "BSD(ref1)" : "BSD");
   }
@@ -1016,6 +1042,10 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     INTERESTING(lDebug ? "BSD(ref14)" : "BSD");
     lmem[_fBSD] = 1;
   }
+  else if (INFILE(_LT_BSDref15)) {
+    INTERESTING(lDebug ? "BSD(ref15)" : "BSD");
+    lmem[_fBSD] = 1;
+  }
   else if (INFILE(_LT_BSDSTYLEref1)) {
     INTERESTING(lDebug ? "BSD-st(1)" : "BSD-style");
     lmem[_fBSD] = 1;
@@ -1236,6 +1266,15 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     INTERESTING("OFL-1.0");
     lmem[_fOFL] = 1;
   }
+  else if (INFILE(_SPDX_OFL_10_no_RFN)) {
+    INTERESTING("OFL-1.0-no-RFN");
+  }
+  else if (INFILE(_SPDX_OFL_10_RFN)) {
+    INTERESTING("OFL-1.0-RFN");
+  }
+  else if (INFILE(_SPDX_OFL_10)) {
+    INTERESTING("OFL-1.0");
+  }
   else if (INFILE(_PHR_OPEN_FONT_V10_1) || INFILE(_PHR_OPEN_FONT_V10_2))
   {
     INTERESTING("OFL-1.0");
@@ -1246,6 +1285,15 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     INTERESTING("OFL-1.1");
     lmem[_fOFL] = 1;
   }
+  else if (INFILE(_SPDX_OFL_11_no_RFN)) {
+    INTERESTING("OFL-1.1-no-RFN");
+  }
+  else if (INFILE(_SPDX_OFL_11_RFN)) {
+    INTERESTING("OFL-1.1-RFN");
+  }
+  else if (INFILE(_SPDX_OFL_11)) {
+    INTERESTING("OFL-1.1");
+  }
   else if (INFILE(_PHR_OPEN_FONT_V11_1) || INFILE(_PHR_OPEN_FONT_V11_2))
   {
     INTERESTING("OFL-1.1");
@@ -1267,18 +1315,18 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
    * It has to be checked before GPL because the license has the reference
    * to GPL license which gives a false positive GPL finding.
    */
-  if (INFILE(_TITLE_COPYLEFT_NEXT_030) && INFILE(_PHR_COPYLEFT_NEXT_PARA1) && INFILE(_PHR_COPYLEFT_NEXT_PARA3)) {
+  if (INFILE(_TITLE_copyleft_next_030) && INFILE(_PHR_copyleft_next_PARA1) && INFILE(_PHR_copyleft_next_PARA3)) {
     INTERESTING("copyleft-next-0.3.0");
     lmem[_mGPL] = 1;
   }
-  else if (INFILE(_TITLE_COPYLEFT_NEXT_031) && INFILE(_PHR_COPYLEFT_NEXT_PARA1) && INFILE(_PHR_COPYLEFT_NEXT_PARA3)) {
+  else if (INFILE(_TITLE_copyleft_next_031) && INFILE(_PHR_copyleft_next_PARA1) && INFILE(_PHR_copyleft_next_PARA3)) {
     INTERESTING("copyleft-next-0.3.1");
     lmem[_mGPL] = 1;
   }
-  else if (INFILE(_PHR_COPYLEFT_NEXT_030) || INFILE(_SPDX_copyleft_next_030)) {
+  else if (INFILE(_PHR_copyleft_next_030) || INFILE(_SPDX_copyleft_next_030)) {
     INTERESTING("copyleft-next-0.3.0");
   }
-  else if (INFILE(_PHR_COPYLEFT_NEXT_031) || INFILE(_SPDX_copyleft_next_031)) {
+  else if (INFILE(_PHR_copyleft_next_031) || INFILE(_SPDX_copyleft_next_031)) {
     INTERESTING("copyleft-next-0.3.1");
   }
   cleanLicenceBuffer();
@@ -1745,7 +1793,7 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     }
     /* checking for FSF */
     if (INFILE(_LT_FSF_1)) {
-      INTERESTING(lDebug ? "FSF(1)" : "FSF");
+      INTERESTING(lDebug ? "FSF(1)" : "FSFULLR");
       lmem[_mLGPL] = 1;
     }
     else if (INFILE(_LT_FSF_2)) {
@@ -1760,6 +1808,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     else if (!lmem[_mGPL] && mCR_FSF() && INFILE(_LT_FSF_5)) {
       INTERESTING(lDebug ? "FSF(5)" : "FSF");
     }
+    else if (!lmem[_mGPL] && INFILE(_LT_FSFUL)) {
+      INTERESTING("FSFUL");
+    }
     else if (!lmem[_mGPL] && NOT_INFILE(_LT_Autoconf_exception_20) && INFILE(_LT_FSFref1)) {
       INTERESTING(lDebug ? "FSF(ref1)" : "FSF");
     }
@@ -2111,6 +2162,10 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
       INTERESTING(lDebug ? "MIT(6)" : "MIT");
       lmem[_mMIT] = 1;
     }
+    else if (!lmem[_mMIT] && INFILE(_LT_HPND_1) && INFILE(_LT_HPND_2)) {
+      INTERESTING("HPND-sell-variant");
+      lmem[_mMIT] = 1;
+    }
     /*
      * _LT_MIT_6 is very similar to _LT_BSD_2 where MIT-CMU licenses
      * are also detected. Following else if is the copy from there.
@@ -2267,7 +2322,7 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   /*
    * ISC License
    */
-  if (INFILE(_PHR_ISC_1) || INFILE(_PHR_ISC_2) || HASTEXT(_URL_ISC, 0)) {
+  if (INFILE(_PHR_ISC_1) || INFILE(_PHR_ISC_2) || HASTEXT(_URL_ISC, REG_EXTENDED)) {
     INTERESTING(lDebug ? "PHR(ISC)" : "ISC");
     lmem[_mISC] = 1;
   }
@@ -2308,7 +2363,10 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     lmem[_fW3C] = 1;
   }
   else if (!lmem[_mNTP] && !lmem[_mPYTHON] && !lmem[_fBSD] && INFILE(_LT_W3C_2)) {
-    if (INFILE(_CR_W3C)) {
+    if (INFILE(_LT_W3C_19980720)) {
+      INTERESTING("W3C-19980720");
+    }
+    else if (INFILE(_CR_W3C)) {
       INTERESTING(lDebug ? "W3C(2)" : "W3C");
     }
     else {
@@ -2317,10 +2375,18 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     lmem[_fW3C] = 1;
   }
   else if (INFILE(_LT_W3C_3)) {
-    if (INFILE(_CR_W3C)) {
+    if (INFILE(_LT_W3C_8)) {
+      if (INFILE(_LT_W3C_20021231)) {
+        INTERESTING("W3C");
+      }
+      else if (INFILE(_LT_W3C_20150513)) {
+        INTERESTING("W3C-20150513");
+      }
+    }
+    else if (INFILE(_CR_W3C)) {
       INTERESTING(lDebug ? "W3C(3)" : "W3C");
     }
-    if (INFILE(_LT_W3Cref4)) {
+    else if (INFILE(_LT_W3Cref4)) {
       INTERESTING(lDebug ? "W3C(ref4)" : "W3C");
     }
     else {
@@ -2359,6 +2425,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     if (INFILE(_CR_W3C)) {
       INTERESTING(lDebug ? "W3C(7)" : "W3C");
     }
+    else if (INFILE(_LT_OGC)) {
+      INTERESTING("OGC");
+    }
     else {
       INTERESTING(lDebug ? "W3C-style(7)" : "W3C-style");
     }
@@ -2393,7 +2462,10 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     lmem[_fW3C] = 1;
   }
   else if (!lmem[_mMIT] && INFILE(_LT_MIT_3)) {
-    if (mCR_MIT()) {
+    if (INFILE(_LT_NTP_0)) {
+      INTERESTING("NTP-0");
+    }
+    else if (mCR_MIT()) {
       INTERESTING(lDebug ? "MIT(8)" : "MIT");
     }
     else {
@@ -2518,6 +2590,12 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (HASTEXT(_TEXT_TU_BERLIN, 0) && HASTEXT(_LT_FREE_87,0)) {
     INTERESTING("TU-Berlin-1.0");
   }
+  else if (INFILE(_LT_NIST_1) && INFILE(_LT_NIST_2)) {
+    INTERESTING(lDebug ? "NIST" : "MIT-style");
+  }
+  else if (INFILE(_LT_FSFAP)) {
+    INTERESTING("FSFAP");
+  }
   cleanLicenceBuffer();
   /*
    * Generic CopyLeft licenses
@@ -2580,7 +2658,12 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   }
   else if (!lmem[_fW3C] && INFILE(_LT_PNG_ZLIB_2)) {
     if (INFILE(_LT_libmng_2007_Clause_1)) {
-      INTERESTING("libmng-2007");
+      if (HASTEXT(_LT_libmng_2007_1, 0)) {
+        INTERESTING("libmng-2007");
+      }
+      else if (INFILE(_LT_libpng_20_Clause_1)) {
+        INTERESTING("libpng-2.0");
+      }
     }
     else {
       INTERESTING(lDebug ? "PNG/ZLIB(2)" : "Libpng");
@@ -3141,6 +3224,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (INFILE(_SPDX_Intel_ACPI)) {
     INTERESTING("Intel-ACPI");
   }
+  else if (INFILE(_LT_ISSL_1) && INFILE(_LT_ISSL_2)) {
+    INTERESTING("ISSL");
+  }
   else if (!lmem[_fREAL] && INFILE(_SPDX_Intel)) {
     INTERESTING("Intel");
   }
@@ -3202,17 +3288,22 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
    * Cisco systems
    */
   if (!lmem[_fPURDUE] && INFILE(_LT_CISCO)) {
-    if (INFILE(_CR_CISCO)) {
-      INTERESTING("Cisco");
-    }
-    else {
-      if (HASTEXT(_LT_PNG_ZLIB_CLAUSE_1, 0) && HASTEXT(_LT_PNG_ZLIB_CLAUSE_2, REG_EXTENDED) && HASTEXT(_LT_PNG_ZLIB_CLAUSE_3, 0)) {
-        INTERESTING("Zlib-style");
+    if (HASTEXT(_LT_PNG_ZLIB_CLAUSE_1, 0) &&
+        HASTEXT(_LT_PNG_ZLIB_CLAUSE_2, REG_EXTENDED) &&
+        HASTEXT(_LT_PNG_ZLIB_CLAUSE_3, 0)) {
+      if (INFILE(_LT_Spencer_86_94_CLAUSE_1) && INFILE(_LT_Spencer_94_CLAUSE_2)) {
+        INTERESTING("Spencer-94");
       }
       else {
-        INTERESTING("Cisco-style");
+        INTERESTING("Zlib-style");
       }
     }
+    else if (INFILE(_CR_CISCO)) {
+      INTERESTING("Cisco");
+    }
+    else {
+      INTERESTING("Cisco-style");
+    }
   }
   cleanLicenceBuffer();
   /*
@@ -3875,13 +3966,6 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     INTERESTING("Ascender-EULA");
   }
   cleanLicenceBuffer();
-  /*
-   * JPNIC
-   */
-  if (HASTEXT(_TEXT_JPNIC, 0) && INFILE(_LT_JPNIC)) {
-    INTERESTING("JPNIC");
-  }
-  cleanLicenceBuffer();
   /*
    * ADAPTEC
    */
@@ -3896,21 +3980,26 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
    * Artistic and Perl
    */
   if (INFILE(_LT_PERL_1)) {
-    INTERESTING(lDebug ? "Artistic(Perl#1)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl#1)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl#1)" : "GPL-1.0+");
     }
   }
   else if (INFILE(_LT_PERL_2)) {
-    INTERESTING(lDebug ? "Artistic(Perl#2)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl#2)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl#2)" : "GPL-1.0+");
     }
   }
   else if (INFILE(_LT_PERL_3)) {
-    if (!lmem[_fOPENLDAP] && !TRYGROUP(famOPENLDAP)) {
+    if (INFILE(_LT_Spencer_86_94_CLAUSE_1) &&
+        HASTEXT(_LT_PNG_ZLIB_CLAUSE_1, 0) &&
+        HASTEXT(_LT_PNG_ZLIB_CLAUSE_2, REG_EXTENDED)) {
+      INTERESTING("Spencer-86");
+    }
+    else if (!lmem[_fOPENLDAP] && !TRYGROUP(famOPENLDAP)) {
       INTERESTING(lDebug ? "Artistic(Perl#3)" : "Artistic-1.0");
     }
   }
@@ -3919,35 +4008,35 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
    * Larry Wall and the documented licensing terms of "perl"
    */
   else if (INFILE(_LT_PERLref1)) {
-    INTERESTING(lDebug ? "Artistic(Perl-ref1)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl-ref1)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl-ref1)" : "GPL-1.0+");
     }
   }
   else if (PERL_INFILE(_LT_PERLref2)) {
-    INTERESTING(lDebug ? "Artistic(Perl-ref2)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl-ref2)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl-ref2)" : "GPL-1.0+");
     }
   }
   else if (INFILE(_LT_PERLref3)) {
-    INTERESTING(lDebug ? "Artistic(Perl-ref3)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl-ref3)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl-ref3)" : "GPL-1.0+");
     }
   }
   else if (INFILE(_LT_PERLref4)) {
-    INTERESTING(lDebug ? "Artistic(Perl-ref4)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl-ref4)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl-ref4)" : "GPL-1.0+");
     }
   }
   else if (INFILE(_LT_PERLref5)) {
-    INTERESTING(lDebug ? "Artistic(Perl-ref5)" : "Artistic-1.0");
+    INTERESTING(lDebug ? "Artistic(Perl-ref5)" : "Artistic-1.0-Perl");
     if (!lmem[_mGPL]) {
       INTERESTING("Dual-license");
       INTERESTING(lDebug ? "GPL(Perl-ref5)" : "GPL-1.0+");
@@ -3970,6 +4059,9 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
       if (INFILE(_LT_ART_V2) || INFILE(_TITLE_ART20)) {
         INTERESTING(lDebug ? "Artistic(v2.0#1)" : "Artistic-2.0");
       }
+      else if (INFILE(_PHR_Artistic_20)) {
+        INTERESTING("Artistic-2.0");
+      }
       else if (INFILE(_TITLE_NBPL_V10)) {
         INTERESTING("NBPL-1.0");
         lmem[_fARTISTIC] = 1;
@@ -4277,8 +4369,11 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   /*
    * Vim license
    */
-  if ((INFILE(_LT_VIM_1) || INFILE(_LT_VIM_2)) && INFILE(_TITLE_VIM)) {
-    INTERESTING("VIM");
+  if ((INFILE(_LT_Vim_1) || INFILE(_LT_Vim_2)) && INFILE(_TITLE_Vim)) {
+    INTERESTING("Vim");
+  }
+  else if (INFILE(_PHR_Vim)) {
+    INTERESTING("Vim");
   }
   cleanLicenceBuffer();
   /*
@@ -4497,19 +4592,8 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   /*
    * RSA Security, Inc.
    */
-  if (INFILE(_CR_RSA)) {
-    if (INFILE(_LT_RSA_1)) {
-      INTERESTING(lDebug ? "RSA(1)" : "RSA-Security");
-    }
-    else if (INFILE(_LT_RSA_2)) {
-      INTERESTING(lDebug ? "RSA(2)" : "RSA-Security");
-    }
-  }
-  else if (INFILE(_LT_RSA_3)) {
-    INTERESTING(lDebug ? "RSA(3)" : "RSA-Security");
-  }
-  else if (INFILE(_LT_RSA_4)) {
-    INTERESTING(lDebug ? "RSA(4)" : "RSA-Security");
+  if (INFILE(_LT_RSA_4)) {
+    INTERESTING(lDebug ? "RSA(4)" : "RSA-MD");
   }
   else if (INFILE(_LT_RSA_5)) {
     INTERESTING(lDebug ? "RSA(5)" : "RSA-DNS");
@@ -4517,6 +4601,17 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (INFILE (_LT_RSA_CRYPTOKI_1) && INFILE(_LT_RSA_CRYPTOKI_2)){
     INTERESTING("RSA-Cryptoki");
   }
+  else if (INFILE(_LT_RSA_3)) {
+    INTERESTING(lDebug ? "RSA(3)" : "RSA-Security");
+  }
+  else if (INFILE(_CR_RSA)) {
+    if (INFILE(_LT_RSA_1)) {
+      INTERESTING(lDebug ? "RSA(1)" : "RSA-Security");
+    }
+    else if (INFILE(_LT_RSA_2)) {
+      INTERESTING(lDebug ? "RSA(2)" : "RSA-Security");
+    }
+  }
   cleanLicenceBuffer();
   /* Some licenses only deal with fonts */
   if (HASTEXT(_TEXT_FONT, 0)) {
@@ -4768,7 +4863,7 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   /*
    * DARPA (Defense Advanved Research Projects Agency)
    */
-  if (INFILE(_LT_DARPA_COUGAAR)) {
+  if (INFILE(_LT_DARPA_COUGAAR_1)) {
     INTERESTING("DARPA-Cougaar");
   }
   else if (INFILE(_LT_DARPA)) {
@@ -6325,6 +6420,45 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
     INTERESTING("unboundID-ldap-sdk");
   }
   cleanLicenceBuffer();
+  /*
+   * BlueOak-1.0.0
+   */
+  if (INFILE(_TITLE_BlueOak_100) || URL_INFILE(_URL_BlueOak_100)) {
+    INTERESTING("BlueOak-1.0.0");
+  }
+  else if (INFILE(_LT_BlueOak_100_Purpose) && INFILE(_LT_BlueOak_100_Acceptance)) {
+    INTERESTING("BlueOak-1.0.0");
+  }
+  cleanLicenceBuffer();
+  /*
+   * CERN-OHL
+   */
+  if (INFILE(_TITLE_CERN_OHL_11)) {
+    INTERESTING("CERN-OHL-1.1");
+  }
+  else if (INFILE(_TITLE_CERN_OHL_12)) {
+    INTERESTING("CERN-OHL-1.2");
+  }
+  cleanLicenceBuffer();
+  /*
+   * MulanPSL
+   */
+  if (INFILE(_TITLE_MulanPSL_10) || URL_INFILE(_URL_MulanPSL) || INFILE(_LT_MulanPSL_10)) {
+    INTERESTING("MulanPSL-1.0");
+  }
+  cleanLicenceBuffer();
+  /*
+   * SSH
+   */
+  if (INFILE(_LT_FREE_72) && HASTEXT(_LT_SSH, REG_EXTENDED)) {
+    if (INFILE(_LT_SSH_OpenSSH)) {
+      INTERESTING("SSH-OpenSSH");
+    }
+    else {
+      INTERESTING("SSH-short");
+    }
+  }
+  cleanLicenceBuffer();
 
   SPDXREF();
   cleanLicenceBuffer();
@@ -6542,7 +6676,7 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   else if (INFILE(_LT_DUAL_LICENSE_48)) {
     MEDINTEREST(lDebug ? "Dual-license(48)" : "Dual-license");
   }
-  else if (INFILE(_LT_DUAL_LICENSE_49)) {
+  else if (HASTEXT(_LT_DUAL_LICENSE_49, REG_EXTENDED)) {
     MEDINTEREST(lDebug ? "Dual-license(49)" : "Dual-license");
   }
   cleanLicenceBuffer();
@@ -6772,12 +6906,13 @@ char *parseLicenses(char *filetext, int size, scanres_t *scp,
   /* not public domain */
   if (HASTEXT(_LT_PUBDOM_NOTclaim, REG_EXTENDED)) {
     if (INFILE(_LT_PUBDOM_CC)) {
-      INTERESTING(lDebug ? "Pubdom(CC)" : LS_PD_CLM);
+      INTERESTING(lDebug ? "Pubdom(CC)" : "CC-PDDC");
+      pd = 1;
     }
     else {
       INTERESTING(LS_NOT_PD);
+      pd = 0;
     }
-    pd = 0;
   }
   cleanLicenceBuffer();
   /* LIBGCJ license */
@@ -7316,7 +7451,7 @@ char *aslVersion(char *filetext, int size, int isML, int isPS)
     lstr = (lDebug ? "Apache(2.0#2)" : "Apache-2.0");
     lmem[_mAPACHE] = 1;
   }
-  else if (INFILE(_PHR_Apache_20_ref4) || INFILE(_PHR_Apache_20_ref5) || INFILE(_PHR_Apache_20_ref6)) {
+  else if (INFILE(_PHR_Apache_20_ref4) || INFILE(_PHR_Apache_20_ref5) || INFILE(_PHR_Apache_20_ref6) || INFILE(_PHR_Apache_20_ref7)) {
     lstr = (lDebug ? "Apache(2.0#3)" : "Apache-2.0");
     lmem[_mAPACHE] = 1;
   }
@@ -7661,7 +7796,7 @@ char *pythonVersion(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_TITLE_PYTHON23)) {
     lstr = "Python-2.3";
   }
-  else if (INFILE(_TITLE_PYTHON2)) {
+  else if (INFILE(_TITLE_PYTHON_20_1) || INFILE(_TITLE_PYTHON_20_2)) {
     lstr = "Python-2.0";
   }
   else {
@@ -8166,20 +8301,47 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
 #endif  /* PROC_TRACE */
   /* */
 
-  if (INFILE(_PHR_GPL21_OR_LATER) && !HASTEXT(_LT_IGNORE_CLAUSE, REG_EXTENDED)) {
-    lstr = "GPL-2.1+[sic]";
+  /*
+   * GPL-3.0 cases
+   */
+  if (GPL_INFILE(_PHR_GPL3_OR_LATER_ref2)
+      || GPL_INFILE(_PHR_GPL3_OR_LATER_ref3)
+      || GPL_INFILE(_PHR_GPL3_OR_LATER)
+      || GPL_INFILE(_PHR_GPL3_OR_LATER_ref1)
+      || HASTEXT(_SPDX_GPL_30plus, REG_EXTENDED)
+      || HASTEXT(_PHR_GPL_30plus, REG_EXTENDED))
+  {
+    if (!HASTEXT(_LT_IGNORE_CLAUSE, REG_EXTENDED)) {
+      lstr = "GPL-3.0+";
+      if (INFILE(_PHR_GPL2_OR_LATER_1)) {
+        lstr = "GPL-2.0+,GPL-3.0+";
+      }
+    }
   }
-  else if ((GPL_INFILE(_PHR_GPL3_ONLY)
+  else if (GPL_INFILE(_PHR_FSF_V3_ONLY)
+      || GPL_INFILE(_PHR_GPL3_ONLY)
       || INFILE(_FILE_GPLv3)
       || GPL_INFILE(_PHR_GPL3_ONLY_ref1)
-      || GPL_INFILE(_PHR_GPL3_ONLY_ref2))
-      && INFILE(_PHR_GPL2_OR_LATER_1))
-  {
-    lstr = "GPL-2.0+,GPL-3.0";
+      || GPL_INFILE(_PHR_GPL3_ONLY_ref2)) {
+    lstr = lDebug ? "GPL-v3(#2)" : "GPL-3.0";
+    if (INFILE(_PHR_GPL2_OR_LATER_1))
+    {
+      lstr = "GPL-2.0+,GPL-3.0";
+    }
+  }
+  else if (NY_INFILE(_TEXT_GPLV3) && NOT_INFILE(_TEXT_GPLV3_FOOTNOTE) &&
+      HASREGEX(_TEXT_GPLV3, filetext)) {
+    lstr = lDebug ? "GPL-v3(#3)" : "GPL-3.0";
   }
+  /*
+   * GPL-2.0 cases
+   */
   else if (HASTEXT(_LT_GPL_V2_NAMED_later, REG_EXTENDED) || INFILE(_TITLE_GPL2_ref1_later)) {
     lstr = lDebug ? "GPLV2+(named)" : "GPL-2.0+";
   }
+  else if (HASTEXT(_SPDX_GPL_20_or_later, REG_EXTENDED)) {
+    lstr = lDebug ? "GPL-2.0+(SPDX)" : "GPL-2.0+";
+  }
   else if (INFILE(_PHR_GPL2_OR_LATER_1)) {
     if (INFILE(_TITLE_GPL_KDE)) {
       lstr = "GPL-2.0+KDEupgradeClause";
@@ -8187,10 +8349,6 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
     else if (INFILE(_PHR_GPL2_ONLY_2) || INFILE(_PHR_GPL2_ONLY_3)) {
       lstr = "GPL-2.0";
     }
-    else if (GPL_INFILE(_PHR_GPL3_OR_LATER_ref2) || GPL_INFILE(_PHR_GPL3_OR_LATER_ref3)
-      || GPL_INFILE(_PHR_GPL3_OR_LATER) || GPL_INFILE(_PHR_GPL3_OR_LATER_ref1)) {
-      lstr = "GPL-2.0+,GPL-3.0+";
-    }
     else if (!HASTEXT(_LT_IGNORE_CLAUSE, REG_EXTENDED)) {
       lstr = lDebug ? "PHR(GPL2_OR_LATER#3)" : "GPL-2.0+";
     }
@@ -8198,6 +8356,9 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
       lstr = lDebug ? "GPL-2.0(title)" : "GPL-2.0";
     }
   }
+  else if (HASTEXT(_SPDX_GPL_20, REG_EXTENDED)) {
+    lstr = lDebug ? "GPL-2.0(SPDX)" : "GPL-2.0";
+  }
   else if (INFILE(_PHR_GPL2_ONLY_4)) {
     lstr = "GPL-2.0";
   }
@@ -8207,28 +8368,19 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
   else if (GPL_INFILE(_PHR_GPL2_OR_GPL3)) {
     lstr = "GPL-2.0,GPL-3.0";
   }
-  else if (GPL_INFILE(_PHR_GPL3_OR_LATER_ref2)
-      || GPL_INFILE(_PHR_GPL3_OR_LATER_ref3)
-      || GPL_INFILE(_PHR_GPL3_OR_LATER)
-      || GPL_INFILE(_PHR_GPL3_OR_LATER_ref1)
-      || HASTEXT(_SPDX_GPL_30plus, REG_EXTENDED)
-      || HASTEXT(_PHR_GPL_30plus, REG_EXTENDED))
-  {
-    if (!HASTEXT(_LT_IGNORE_CLAUSE, REG_EXTENDED)) {
-      lstr = "GPL-3.0+";
-    }
-  }
-  else if (GPL_INFILE(_PHR_FSF_V3_ONLY) || GPL_INFILE(_PHR_GPL3_ONLY) ||
-      INFILE(_FILE_GPLv3) || GPL_INFILE(_PHR_GPL3_ONLY_ref1) || GPL_INFILE(_PHR_GPL3_ONLY_ref2)) {
-    lstr = lDebug ? "GPL-v3(#2)" : "GPL-3.0";
-  }
-  else if (INFILE(_PHR_FSF_V21_ONLY) || INFILE(_PHR_GPL21_ONLY)) {
-    lstr = lDebug ? "GPL-v2.1[sic]" : "GPL-2.1[sic]";
-  }
   else if (INFILE(_PHR_FSF_V2_ONLY) || INFILE(_PHR_GPL2_ONLY) || INFILE(_PHR_GPL2_ONLY_1) ||
       INFILE(_FILE_GPLv2) || INFILE(_LT_GPL_V2_NAMED)) {
     lstr = lDebug ? "GPL-v2(#2)" : "GPL-2.0";
   }
+  else if (INFILE(_LT_GPL_V2_ref5)) {
+    lstr = lDebug ? "GPL-2.0(ref5)" : "GPL-2.0";
+  }
+  else if (NY_INFILE(_TEXT_GPLV2)) {
+    lstr = lDebug ? "GPL-v2(#3)" : "GPL-2.0";
+  }
+  /*
+   * GPL-1.0 cases
+   */
   else if (GPL_INFILE(_PHR_FSF_V1_OR_LATER)
       || INFILE(_PHR_GPL1_OR_LATER)
       || HASTEXT(_SPDX_GPL_10plus, REG_EXTENDED)
@@ -8247,19 +8399,9 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
   else if (URL_INFILE(_URL_CCGPL_V2)) {
     lstr = "GPL-2.0";
   }
-  else if (INFILE(_LT_GPL_V2_ref5)) {
-    lstr = lDebug ? "GPL-2.0(ref5)" : "GPL-2.0";
-  }
   else if (INFILE(_LT_CC_GPL) || INFILE(_TITLE_CC_GPL)) {
     lstr = lDebug ? "GPL(CC_GPL)" : "GPL";
   }
-  else if (NY_INFILE(_TEXT_GPLV3) && NOT_INFILE(_TEXT_GPLV3_FOOTNOTE) &&
-      HASREGEX(_TEXT_GPLV3, filetext)) {
-    lstr = lDebug ? "GPL-v3(#3)" : "GPL-3.0";
-  }
-  else if (NY_INFILE(_TEXT_GPLV2)) {
-    lstr = lDebug ? "GPL-v2(#3)" : "GPL-2.0";
-  }
   else if (NY_INFILE(_TEXT_GPLV1) &&
       HASREGEX(_TEXT_GPLV1, filetext)) {
     lstr = lDebug ? "GPL-v1(#3)" : "GPL-1.0";
@@ -8267,9 +8409,6 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
   else if (HASTEXT(_SPDX_GPL_10, REG_EXTENDED)) {
     lstr = lDebug ? "GPL-1.0(SPDX)" : "GPL-1.0";
   }
-  else if (HASTEXT(_SPDX_GPL_20, REG_EXTENDED)) {
-    lstr = lDebug ? "GPL-2.0(SPDX)" : "GPL-2.0";
-  }
   else if (HASTEXT(_SPDX_GPL_30, REG_EXTENDED)) {
     lstr = lDebug ? "GPL-3.0(SPDX)" : "GPL-3.0";
   }
@@ -8292,6 +8431,15 @@ char *gplVersion(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_TITLE_MODULE_LICENSE_GPL)) {
     lstr = lDebug ? "GPL(linux-kernel)" : "GPL";
   }
+  /*
+   * Finally let's see if there is a type error in license version
+   */
+  else if (INFILE(_PHR_GPL21_OR_LATER) && !HASTEXT(_LT_IGNORE_CLAUSE, REG_EXTENDED)) {
+    lstr = "GPL-2.1+[sic]";
+  }
+  else if (INFILE(_PHR_FSF_V21_ONLY) || INFILE(_PHR_GPL21_ONLY)) {
+    lstr = lDebug ? "GPL-v2.1[sic]" : "GPL-2.1[sic]";
+  }
   /*
    * Special case, HACK: "Debian packaging ... licensed under GPL"
    *****
@@ -8490,8 +8638,9 @@ char *ccVersion(char *filetext, int size, int isML, int isPS)
   else if (URL_INFILE(_URL_CC0)) {
     lstr = lDebug ? "CC0(URL)" : "CC0-1.0";
   }
-  else if (URL_INFILE(_URL_CC_PD)) {
-    lstr = lDebug ? "CC(PD)" : "Public-domain";
+  else if (URL_INFILE(_URL_CC_PDDC)) {
+    lstr = lDebug ? "CC(PD)" : "CC-PDDC";
+    pd = 1;
   }
   else if (INFILE(_TITLE_CCPL)) {
     INTERESTING("CCPL");
@@ -9932,7 +10081,7 @@ int checkPublicDomain(char *filetext, int size, int score, int kwbm,
   }
   ret = 0;        /* default answer is "no" */
   if (INFILE(_LT_PUBDOM_CC)) {
-    INTERESTING(lDebug ? "Pubdom(CC)" : LS_PD_CLM);
+    INTERESTING(lDebug ? "Pubdom(CC)" : "CC-PDDC");
     ret = 1;
   } else if (INFILE(_LT_PUBDOM_ODC)) {
     INTERESTING(lDebug ? "Pubdom(ODC)" : LS_PD_CLM);
@@ -9989,7 +10138,12 @@ int checkPublicDomain(char *filetext, int size, int score, int kwbm,
     INTERESTING(lDebug ? "Pubdom(9)" : LS_PD_CLM);
     ret = 1;
   } else if (INFILE(_LT_PUBDOM_10)) {
-    INTERESTING(lDebug ? "Pubdom(10)" : LS_PD_CLM);
+      if (INFILE(_LT_blessing)) {
+        INTERESTING(lDebug ? "Pubdom(10)" : "blessing");
+      }
+      else {
+        INTERESTING(lDebug ? "Pubdom(10)" : LS_PD_CLM);
+      }
     ret = 1;
   } else if (INFILE(_URL_PUBDOM)) {
     INTERESTING(lDebug ? "Pubdom(URL)" : LS_PD_CLM);
@@ -10394,6 +10548,12 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_BitTorrent_11)) {
     INTERESTING("BitTorrent-1.1");
   }
+  if (INFILE(_SPDX_blessing)) {
+    INTERESTING("blessing");
+  }
+  if (INFILE(_SPDX_BlueOak_100)) {
+    INTERESTING("BlueOak-1.0.0");
+  }
   if (INFILE(_SPDX_BSL_10)) {
     INTERESTING("BSL-1.0");
   }
@@ -10412,6 +10572,24 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_Caldera)) {
     INTERESTING("Caldera");
   }
+  if (INFILE(_SPDX_CC_PDDC)) {
+    INTERESTING("CC-PDDC");
+  }
+  if (INFILE(_SPDX_CERN_OHL_P_20)) {
+    INTERESTING("CERN-OHL-P-2.0");
+  }
+  else if (INFILE(_SPDX_CERN_OHL_S_20)) {
+    INTERESTING("CERN-OHL-S-2.0");
+  }
+  else if (INFILE(_SPDX_CERN_OHL_W_20)) {
+    INTERESTING("CERN-OHL-W-2.0");
+  }
+  else if (INFILE(_SPDX_CERN_OHL_12)) {
+    INTERESTING("CERN-OHL-1.2");
+  }
+  else if (INFILE(_SPDX_CERN_OHL_11)) {
+    INTERESTING("CERN-OHL-1.1");
+  }
   if (INFILE(_SPDX_ClArtistic)) {
     INTERESTING("ClArtistic");
   }
@@ -10427,6 +10605,12 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_CPL_10)) {
     INTERESTING("CPL-1.0");
   }
+  if (INFILE(_SPDX_CAL_10_Combined_Work_Exception)) {
+    INTERESTING("CAL-1.0-Combined-Work-Exception");
+  }
+  else if (INFILE(_SPDX_CAL_10)) {
+    INTERESTING("CAL-1.0");
+  }
   if (INFILE(_SPDX_CATOSL_11)) {
     INTERESTING("CATOSL-1.1");
   }
@@ -10442,6 +10626,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_CC_BY_25)) {
     INTERESTING("CC-BY-2.5");
   }
+  else if (INFILE(_SPDX_CC_BY_30_AT)) {
+    INTERESTING("CC-BY-3.0-AT");
+  }
   else if (INFILE(_SPDX_CC_BY_30)) {
     INTERESTING("CC-BY-3.0");
   }
@@ -10487,6 +10674,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_CC_BY_NC_ND_25)) {
     INTERESTING("CC-BY-NC-ND-2.5");
   }
+  else if (INFILE(_SPDX_CC_BY_NC_ND_30_IGO)) {
+    INTERESTING("CC-BY-NC-ND-3.0-IGO");
+  }
   else if (INFILE(_SPDX_CC_BY_NC_ND_30)) {
     INTERESTING("CC-BY-NC-ND-3.0");
   }
@@ -10517,6 +10707,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_CC_BY_SA_25)) {
     INTERESTING("CC-BY-SA-2.5");
   }
+  else if (INFILE(_SPDX_CC_BY_SA_30_AT)) {
+    INTERESTING("CC-BY-SA-3.0-AT");
+  }
   else if (INFILE(_SPDX_CC_BY_SA_30)) {
     INTERESTING("CC-BY-SA-3.0");
   }
@@ -10589,9 +10782,15 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_Entessa)) {
     INTERESTING("Entessa");
   }
+  if (INFILE(_SPDX_EPICS)) {
+    INTERESTING("EPICS");
+  }
   if (INFILE(_SPDX_ErlPL_11)) {
     INTERESTING("ErlPL-1.1");
   }
+  if (INFILE(_SPDX_etalab_20)) {
+    INTERESTING("etalab-2.0");
+  }
   if (INFILE(_SPDX_EUDatagrid)) {
     INTERESTING("EUDatagrid");
   }
@@ -10634,7 +10833,19 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_Glulxe)) {
     INTERESTING("Glulxe");
   }
-  if (INFILE(_SPDX_GFDL_11_or_later)
+  if (INFILE(_SPDX_GFDL_11_invariants_or_later)) {
+    INTERESTING("GFDL-1.1-invariants-or-later");
+  }
+  else if (INFILE(_SPDX_GFDL_11_no_invariants_or_later)) {
+    INTERESTING("GFDL-1.1-no-invariants-or-later");
+  }
+  else if (INFILE(_SPDX_GFDL_11_invariants_only)) {
+    INTERESTING("GFDL-1.1-invariants-only");
+  }
+  else if (INFILE(_SPDX_GFDL_11_no_invariants_only)) {
+    INTERESTING("GFDL-1.1-no-invariants-only");
+  }
+  else if (INFILE(_SPDX_GFDL_11_or_later)
       || HASTEXT(_SPDX_GFDL_11plus, REG_EXTENDED)
       || HASTEXT(_PHR_GFDL_11plus, REG_EXTENDED))
   {
@@ -10643,6 +10854,18 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_GFDL_11)) {
     INTERESTING("GFDL-1.1");
   }
+  else if (INFILE(_SPDX_GFDL_12_invariants_or_later)) {
+    INTERESTING("GFDL-1.2-invariants-or-later");
+  }
+  else if (INFILE(_SPDX_GFDL_12_no_invariants_or_later)) {
+    INTERESTING("GFDL-1.2-no-invariants-or-later");
+  }
+  else if (INFILE(_SPDX_GFDL_12_invariants_only)) {
+    INTERESTING("GFDL-1.2-invariants-only");
+  }
+  else if (INFILE(_SPDX_GFDL_12_no_invariants_only)) {
+    INTERESTING("GFDL-1.2-no-invariants-only");
+  }
   else if (INFILE(_SPDX_GFDL_12_or_later)
       || HASTEXT(_SPDX_GFDL_12plus, REG_EXTENDED)
       || HASTEXT(_PHR_GFDL_12plus, REG_EXTENDED))
@@ -10652,6 +10875,18 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_GFDL_12)) {
     INTERESTING("GFDL-1.2");
   }
+  else if (INFILE(_SPDX_GFDL_13_invariants_or_later)) {
+    INTERESTING("GFDL-1.3-invariants-or-later");
+  }
+  else if (INFILE(_SPDX_GFDL_13_no_invariants_or_later)) {
+    INTERESTING("GFDL-1.3-no-invariants-or-later");
+  }
+  else if (INFILE(_SPDX_GFDL_13_invariants_only)) {
+    INTERESTING("GFDL-1.3-invariants-only");
+  }
+  else if (INFILE(_SPDX_GFDL_13_no_invariants_only)) {
+    INTERESTING("GFDL-1.3-no-invariants-only");
+  }
   else if (INFILE(_SPDX_GFDL_13_or_later)
       || HASTEXT(_SPDX_GFDL_13plus, REG_EXTENDED)
       || HASTEXT(_PHR_GFDL_13plus, REG_EXTENDED))
@@ -10661,6 +10896,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_GFDL_13)) {
     INTERESTING("GFDL-1.3");
   }
+  if (INFILE(_SPDX_GLWTPL)) {
+    INTERESTING("GLWTPL");
+  }
   if (INFILE(_SPDX_gnuplot)) {
     INTERESTING("gnuplot");
   }
@@ -10670,7 +10908,13 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_HaskellReport)) {
     INTERESTING("HaskellReport");
   }
-  if (INFILE(_SPDX_HPND)) {
+  if (INFILE(_SPDX_Hippocratic_21)) {
+    INTERESTING("Hippocratic-2.1");
+  }
+  if (INFILE(_SPDX_HPND_sell_variant)) {
+    INTERESTING("HPND-sell-variant");
+  }
+  else if (INFILE(_SPDX_HPND)) {
     INTERESTING("HPND");
   }
   if (INFILE(_SPDX_IBM_pibs)) {
@@ -10706,6 +10950,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_JasPer_20)) {
     INTERESTING("JasPer-2.0");
   }
+  if (INFILE(_SPDX_JPNIC)) {
+    INTERESTING("JPNIC");
+  }
   if (INFILE(_SPDX_JSON)) {
     INTERESTING("JSON");
   }
@@ -10718,9 +10965,15 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_LGPLLR)) {
     INTERESTING("LGPLLR");
   }
-  if (INFILE(_SPDX_Libpng)) {
+  if (INFILE(_SPDX_libpng_20)) {
+    INTERESTING("libpng-2.0");
+  }
+  else if (INFILE(_SPDX_Libpng)) {
     INTERESTING("Libpng");
   }
+  if (INFILE(_SPDX_libselinux_10)) {
+    INTERESTING("libselinux-1.0");
+  }
   if (INFILE(_SPDX_libtiff)) {
     INTERESTING("libtiff");
   }
@@ -10787,6 +11040,12 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_mpich2)) {
     INTERESTING("mpich2");
   }
+  if (INFILE(_SPDX_MulanPSL_20)) {
+    INTERESTING("MulanPSL-2.0");
+  }
+  else if (INFILE(_SPDX_MulanPSL_10)) {
+    INTERESTING("MulanPSL-1.0");
+  }
   if (INFILE(_SPDX_Multics)) {
     INTERESTING("Multics");
   }
@@ -10802,6 +11061,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_NBPL_10)) {
     INTERESTING("NBPL-1.0");
   }
+  if (INFILE(_SPDX_NCGL_UK_20)) {
+    INTERESTING("NCGL-UK-2.0");
+  }
   if (INFILE(_SPDX_Net_SNMP)) {
     INTERESTING("Net-SNMP");
   }
@@ -10811,6 +11073,12 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_NGPL)) {
     INTERESTING("NGPL");
   }
+  if (INFILE(_SPDX_NIST_PD_fallback)) {
+    INTERESTING("NIST-PD-fallback");
+  }
+  else if (INFILE(_SPDX_NIST_PD)) {
+    INTERESTING("NIST-PD");
+  }
   if (INFILE(_SPDX_NOSL)) {
     INTERESTING("NOSL");
   }
@@ -10841,18 +11109,27 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_NRL)) {
     INTERESTING("NRL");
   }
-  if (INFILE(_SPDX_NTP)) {
+  if (INFILE(_SPDX_NTP_0)) {
+    INTERESTING("NTP-0");
+  }
+  else if (INFILE(_SPDX_NTP)) {
     INTERESTING("NTP");
   }
   if (INFILE(_SPDX_Nunit)) {
     INTERESTING("Nunit");
   }
+  if (INFILE(_SPDX_O_UDA_10)) {
+    INTERESTING("O-UDA-1.0");
+  }
   if (INFILE(_SPDX_OCLC_20)) {
     INTERESTING("OCLC-2.0");
   }
   if (INFILE(_SPDX_ODbL_10)) {
     INTERESTING("ODbL-1.0");
   }
+  if (INFILE(_SPDX_OGC_10)) {
+    INTERESTING("OGC-1.0");
+  }
   if (INFILE(_SPDX_PDDL_10)) {
     INTERESTING("PDDL-1.0");
   }
@@ -10862,6 +11139,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_ODC_By_10)) {
     INTERESTING("ODC-By-1.0");
   }
+  if (INFILE(_SPDX_OGL_Canada_20)) {
+    INTERESTING("OGL-Canada-2.0");
+  }
   if (INFILE(_SPDX_OGL_UK_10)) {
     INTERESTING("OGL-UK-1.0");
   }
@@ -10946,6 +11226,12 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_OSET_PL_21)) {
     INTERESTING("OSET-PL-2.1");
   }
+  if (INFILE(_SPDX_Parity_700)) {
+    INTERESTING("Parity-7.0.0");
+  }
+  else if (INFILE(_SPDX_Parity_600)) {
+    INTERESTING("Parity-6.0.0");
+  }
   if (INFILE(_SPDX_PHP_301)) {
     INTERESTING("PHP-3.01");
   }
@@ -10955,9 +11241,18 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_Plexus)) {
     INTERESTING("Plexus");
   }
+  if (INFILE(_SPDX_PolyForm_Noncommercial_100)) {
+    INTERESTING("PolyForm-Noncommercial-1.0.0");
+  }
+  else if (INFILE(_SPDX_PolyForm_Small_Business_100)) {
+    INTERESTING("PolyForm-Small-Business-1.0.0");
+  }
   if (INFILE(_SPDX_PostgreSQL)) {
     INTERESTING("PostgreSQL");
   }
+  if (INFILE(_SPDX_PSF_20)) {
+    INTERESTING("PSF-2.0");
+  }
   if (INFILE(_SPDX_psfrag)) {
     INTERESTING("psfrag");
   }
@@ -11003,6 +11298,12 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_Saxpath)) {
     INTERESTING("Saxpath");
   }
+  if (INFILE(_SPDX_SHL_051)) {
+    INTERESTING("SHL-0.51");
+  }
+  else if (INFILE(_SPDX_SHL_05)) {
+    INTERESTING("SHL-0.5");
+  }
   if (INFILE(_SPDX_SCEA)) {
     INTERESTING("SCEA");
   }
@@ -11027,12 +11328,6 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_SGI_B_20)) {
     INTERESTING("SGI-B-2.0");
   }
-  if (INFILE(_SPDX_OFL_10)) {
-    INTERESTING("OFL-1.0");
-  }
-  if (INFILE(_SPDX_OFL_11)) {
-    INTERESTING("OFL-1.1");
-  }
   if (INFILE(_SPDX_SimPL_20)) {
     INTERESTING("SimPL-2.0");
   }
@@ -11054,6 +11349,15 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_SMLNJ)) {
     INTERESTING("SMLNJ");
   }
+  if (INFILE(_SPDX_SSH_OpenSSH)) {
+    INTERESTING("SSH-OpenSSH");
+  }
+  if (INFILE(_SPDX_SSH_short)) {
+    INTERESTING("SSH-short");
+  }
+  if (INFILE(_SPDX_SSPL_10)) {
+    INTERESTING("SSPL-1.0");
+  }
   if (INFILE(_SPDX_SugarCRM_113)) {
     INTERESTING("SugarCRM-1.1.3");
   }
@@ -11069,6 +11373,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   if (INFILE(_SPDX_Watcom_10)) {
     INTERESTING("Watcom-1.0");
   }
+  if (INFILE(_SPDX_TAPR_OHL_10)) {
+    INTERESTING("TAPR-OHL-1.0");
+  }
   if (INFILE(_SPDX_TCP_wrappers)) {
     INTERESTING("TCP-wrappers");
   }
@@ -11090,6 +11397,9 @@ void spdxReference(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_SPDX_TU_Berlin_20)) {
     INTERESTING("TU-Berlin-2.0");
   }
+  if (INFILE(_SPDX_UCL_10)) {
+    INTERESTING("UCL-1.0");
+  }
   if (INFILE(_SPDX_Unicode_DFS_2015)) {
     INTERESTING("Unicode-DFS-2015");
   }
@@ -11275,6 +11585,9 @@ void copyleftExceptions(char *filetext, int size, int isML, int isPS)
   else if (INFILE(_LT_GPL_EXCEPT_5) && INFILE(_LT_Autoconf_exception_30)) {
     INTERESTING("Autoconf-exception-3.0");
   }
+  else if (INFILE(_PHR_Autoconf_exception_30)) {
+    INTERESTING("Autoconf-exception-3.0");
+  }
   else if (INFILE(_LT_Autoconf_exception_3)) {
     INTERESTING("Autoconf-exception");
   }
-- 
2.20.1

